// @commercelayer/sdk/lib/esm/resources/axerve_payments.js
import { ApiResource } from '../resource';
class AxervePayments extends ApiResource {
    async create(resource, params, options) {
        return this.resources.create({ ...resource, type: AxervePayments.TYPE }, params, options);
    }
    async update(resource, params, options) {
        return this.resources.update({ ...resource, type: AxervePayments.TYPE }, params, options);
    }
    async delete(id, options) {
        await this.resources.delete((typeof id === 'string') ? { id, type: AxervePayments.TYPE } : id, options);
    }
    async order(axervePaymentId, params, options) {
        const _axervePaymentId = axervePaymentId.id || axervePaymentId;
        return this.resources.fetch({ type: 'orders' }, `axerve_payments/${_axervePaymentId}/order`, params, options);
    }
    async payment_gateway(axervePaymentId, params, options) {
        const _axervePaymentId = axervePaymentId.id || axervePaymentId;
        return this.resources.fetch({ type: 'payment_gateways' }, `axerve_payments/${_axervePaymentId}/payment_gateway`, params, options);
    }
    async versions(axervePaymentId, params, options) {
        const _axervePaymentId = axervePaymentId.id || axervePaymentId;
        return this.resources.fetch({ type: 'versions' }, `axerve_payments/${_axervePaymentId}/versions`, params, options);
    }
    async _update(id, params, options) {
        return this.resources.update({ id: (typeof id === 'string') ? id : id.id, type: AxervePayments.TYPE, _update: true }, params, options);
    }
    isAxervePayment(resource) {
        return resource.type && (resource.type === AxervePayments.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: AxervePayments.TYPE } : { id: id.id, type: AxervePayments.TYPE };
    }
    type() {
        return AxervePayments.TYPE;
    }
}
AxervePayments.TYPE = 'axerve_payments';
export default AxervePayments;
