// @commercelayer/sdk/lib/esm/resources/carrier_accounts.js
import { ApiResource } from '../resource';
class CarrierAccounts extends ApiResource {
    async market(carrierAccountId, params, options) {
        const _carrierAccountId = carrierAccountId.id || carrierAccountId;
        return this.resources.fetch({ type: 'markets' }, `carrier_accounts/${_carrierAccountId}/market`, params, options);
    }
    async attachments(carrierAccountId, params, options) {
        const _carrierAccountId = carrierAccountId.id || carrierAccountId;
        return this.resources.fetch({ type: 'attachments' }, `carrier_accounts/${_carrierAccountId}/attachments`, params, options);
    }
    async versions(carrierAccountId, params, options) {
        const _carrierAccountId = carrierAccountId.id || carrierAccountId;
        return this.resources.fetch({ type: 'versions' }, `carrier_accounts/${_carrierAccountId}/versions`, params, options);
    }
    isCarrierAccount(resource) {
        return resource.type && (resource.type === CarrierAccounts.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: CarrierAccounts.TYPE } : { id: id.id, type: CarrierAccounts.TYPE };
    }
    type() {
        return CarrierAccounts.TYPE;
    }
}
CarrierAccounts.TYPE = 'carrier_accounts';
export default CarrierAccounts;
