// @commercelayer/sdk/lib/esm/resources/custom_promotion_rules.js
import { ApiResource } from '../resource';
class CustomPromotionRules extends ApiResource {
    async create(resource, params, options) {
        return this.resources.create({ ...resource, type: CustomPromotionRules.TYPE }, params, options);
    }
    async update(resource, params, options) {
        return this.resources.update({ ...resource, type: CustomPromotionRules.TYPE }, params, options);
    }
    async delete(id, options) {
        await this.resources.delete((typeof id === 'string') ? { id, type: CustomPromotionRules.TYPE } : id, options);
    }
    async versions(customPromotionRuleId, params, options) {
        const _customPromotionRuleId = customPromotionRuleId.id || customPromotionRuleId;
        return this.resources.fetch({ type: 'versions' }, `custom_promotion_rules/${_customPromotionRuleId}/versions`, params, options);
    }
    isCustomPromotionRule(resource) {
        return resource.type && (resource.type === CustomPromotionRules.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: CustomPromotionRules.TYPE } : { id: id.id, type: CustomPromotionRules.TYPE };
    }
    type() {
        return CustomPromotionRules.TYPE;
    }
}
CustomPromotionRules.TYPE = 'custom_promotion_rules';
export default CustomPromotionRules;
