// @commercelayer/sdk/lib/esm/resources/delivery_lead_times.js
import { ApiResource } from '../resource';
class DeliveryLeadTimes extends ApiResource {
    async create(resource, params, options) {
        return this.resources.create({ ...resource, type: DeliveryLeadTimes.TYPE }, params, options);
    }
    async update(resource, params, options) {
        return this.resources.update({ ...resource, type: DeliveryLeadTimes.TYPE }, params, options);
    }
    async delete(id, options) {
        await this.resources.delete((typeof id === 'string') ? { id, type: DeliveryLeadTimes.TYPE } : id, options);
    }
    async stock_location(deliveryLeadTimeId, params, options) {
        const _deliveryLeadTimeId = deliveryLeadTimeId.id || deliveryLeadTimeId;
        return this.resources.fetch({ type: 'stock_locations' }, `delivery_lead_times/${_deliveryLeadTimeId}/stock_location`, params, options);
    }
    async shipping_method(deliveryLeadTimeId, params, options) {
        const _deliveryLeadTimeId = deliveryLeadTimeId.id || deliveryLeadTimeId;
        return this.resources.fetch({ type: 'shipping_methods' }, `delivery_lead_times/${_deliveryLeadTimeId}/shipping_method`, params, options);
    }
    async attachments(deliveryLeadTimeId, params, options) {
        const _deliveryLeadTimeId = deliveryLeadTimeId.id || deliveryLeadTimeId;
        return this.resources.fetch({ type: 'attachments' }, `delivery_lead_times/${_deliveryLeadTimeId}/attachments`, params, options);
    }
    async versions(deliveryLeadTimeId, params, options) {
        const _deliveryLeadTimeId = deliveryLeadTimeId.id || deliveryLeadTimeId;
        return this.resources.fetch({ type: 'versions' }, `delivery_lead_times/${_deliveryLeadTimeId}/versions`, params, options);
    }
    isDeliveryLeadTime(resource) {
        return resource.type && (resource.type === DeliveryLeadTimes.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: DeliveryLeadTimes.TYPE } : { id: id.id, type: DeliveryLeadTimes.TYPE };
    }
    type() {
        return DeliveryLeadTimes.TYPE;
    }
}
DeliveryLeadTimes.TYPE = 'delivery_lead_times';
export default DeliveryLeadTimes;
