// @commercelayer/sdk/lib/esm/resources/order_validation_rules.js
import { ApiResource } from '../resource';
class OrderValidationRules extends ApiResource {
    async market(orderValidationRuleId, params, options) {
        const _orderValidationRuleId = orderValidationRuleId.id || orderValidationRuleId;
        return this.resources.fetch({ type: 'markets' }, `order_validation_rules/${_orderValidationRuleId}/market`, params, options);
    }
    async attachments(orderValidationRuleId, params, options) {
        const _orderValidationRuleId = orderValidationRuleId.id || orderValidationRuleId;
        return this.resources.fetch({ type: 'attachments' }, `order_validation_rules/${_orderValidationRuleId}/attachments`, params, options);
    }
    async versions(orderValidationRuleId, params, options) {
        const _orderValidationRuleId = orderValidationRuleId.id || orderValidationRuleId;
        return this.resources.fetch({ type: 'versions' }, `order_validation_rules/${_orderValidationRuleId}/versions`, params, options);
    }
    isOrderValidationRule(resource) {
        return resource.type && (resource.type === OrderValidationRules.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: OrderValidationRules.TYPE } : { id: id.id, type: OrderValidationRules.TYPE };
    }
    type() {
        return OrderValidationRules.TYPE;
    }
}
OrderValidationRules.TYPE = 'order_validation_rules';
export default OrderValidationRules;
