// @commercelayer/sdk/lib/esm/resources/payment_gateways.js
import { ApiResource } from '../resource';
class PaymentGateways extends ApiResource {
    async payment_methods(paymentGatewayId, params, options) {
        const _paymentGatewayId = paymentGatewayId.id || paymentGatewayId;
        return this.resources.fetch({ type: 'payment_methods' }, `payment_gateways/${_paymentGatewayId}/payment_methods`, params, options);
    }
    async versions(paymentGatewayId, params, options) {
        const _paymentGatewayId = paymentGatewayId.id || paymentGatewayId;
        return this.resources.fetch({ type: 'versions' }, `payment_gateways/${_paymentGatewayId}/versions`, params, options);
    }
    isPaymentGateway(resource) {
        return resource.type && (resource.type === PaymentGateways.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: PaymentGateways.TYPE } : { id: id.id, type: PaymentGateways.TYPE };
    }
    type() {
        return PaymentGateways.TYPE;
    }
}
PaymentGateways.TYPE = 'payment_gateways';
export default PaymentGateways;
