// @commercelayer/sdk/lib/esm/resources/price_volume_tiers.js
import { ApiResource } from '../resource';
class PriceVolumeTiers extends ApiResource {
    async create(resource, params, options) {
        return this.resources.create({ ...resource, type: PriceVolumeTiers.TYPE }, params, options);
    }
    async update(resource, params, options) {
        return this.resources.update({ ...resource, type: PriceVolumeTiers.TYPE }, params, options);
    }
    async delete(id, options) {
        await this.resources.delete((typeof id === 'string') ? { id, type: PriceVolumeTiers.TYPE } : id, options);
    }
    async price(priceVolumeTierId, params, options) {
        const _priceVolumeTierId = priceVolumeTierId.id || priceVolumeTierId;
        return this.resources.fetch({ type: 'prices' }, `price_volume_tiers/${_priceVolumeTierId}/price`, params, options);
    }
    async attachments(priceVolumeTierId, params, options) {
        const _priceVolumeTierId = priceVolumeTierId.id || priceVolumeTierId;
        return this.resources.fetch({ type: 'attachments' }, `price_volume_tiers/${_priceVolumeTierId}/attachments`, params, options);
    }
    async versions(priceVolumeTierId, params, options) {
        const _priceVolumeTierId = priceVolumeTierId.id || priceVolumeTierId;
        return this.resources.fetch({ type: 'versions' }, `price_volume_tiers/${_priceVolumeTierId}/versions`, params, options);
    }
    async events(priceVolumeTierId, params, options) {
        const _priceVolumeTierId = priceVolumeTierId.id || priceVolumeTierId;
        return this.resources.fetch({ type: 'events' }, `price_volume_tiers/${_priceVolumeTierId}/events`, params, options);
    }
    isPriceVolumeTier(resource) {
        return resource.type && (resource.type === PriceVolumeTiers.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: PriceVolumeTiers.TYPE } : { id: id.id, type: PriceVolumeTiers.TYPE };
    }
    type() {
        return PriceVolumeTiers.TYPE;
    }
}
PriceVolumeTiers.TYPE = 'price_volume_tiers';
export default PriceVolumeTiers;
