// @commercelayer/sdk/lib/esm/resources/satispay_gateways.js
import { ApiResource } from '../resource';
class SatispayGateways extends ApiResource {
    async create(resource, params, options) {
        return this.resources.create({ ...resource, type: SatispayGateways.TYPE }, params, options);
    }
    async update(resource, params, options) {
        return this.resources.update({ ...resource, type: SatispayGateways.TYPE }, params, options);
    }
    async delete(id, options) {
        await this.resources.delete((typeof id === 'string') ? { id, type: SatispayGateways.TYPE } : id, options);
    }
    async payment_methods(satispayGatewayId, params, options) {
        const _satispayGatewayId = satispayGatewayId.id || satispayGatewayId;
        return this.resources.fetch({ type: 'payment_methods' }, `satispay_gateways/${_satispayGatewayId}/payment_methods`, params, options);
    }
    async versions(satispayGatewayId, params, options) {
        const _satispayGatewayId = satispayGatewayId.id || satispayGatewayId;
        return this.resources.fetch({ type: 'versions' }, `satispay_gateways/${_satispayGatewayId}/versions`, params, options);
    }
    async satispay_payments(satispayGatewayId, params, options) {
        const _satispayGatewayId = satispayGatewayId.id || satispayGatewayId;
        return this.resources.fetch({ type: 'satispay_payments' }, `satispay_gateways/${_satispayGatewayId}/satispay_payments`, params, options);
    }
    isSatispayGateway(resource) {
        return resource.type && (resource.type === SatispayGateways.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: SatispayGateways.TYPE } : { id: id.id, type: SatispayGateways.TYPE };
    }
    type() {
        return SatispayGateways.TYPE;
    }
}
SatispayGateways.TYPE = 'satispay_gateways';
export default SatispayGateways;
