// @commercelayer/sdk/lib/esm/resources/tax_rules.js
import { ApiResource } from '../resource';
class TaxRules extends ApiResource {
    async create(resource, params, options) {
        return this.resources.create({ ...resource, type: TaxRules.TYPE }, params, options);
    }
    async update(resource, params, options) {
        return this.resources.update({ ...resource, type: TaxRules.TYPE }, params, options);
    }
    async delete(id, options) {
        await this.resources.delete((typeof id === 'string') ? { id, type: TaxRules.TYPE } : id, options);
    }
    async manual_tax_calculator(taxRuleId, params, options) {
        const _taxRuleId = taxRuleId.id || taxRuleId;
        return this.resources.fetch({ type: 'manual_tax_calculators' }, `tax_rules/${_taxRuleId}/manual_tax_calculator`, params, options);
    }
    async versions(taxRuleId, params, options) {
        const _taxRuleId = taxRuleId.id || taxRuleId;
        return this.resources.fetch({ type: 'versions' }, `tax_rules/${_taxRuleId}/versions`, params, options);
    }
    isTaxRule(resource) {
        return resource.type && (resource.type === TaxRules.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: TaxRules.TYPE } : { id: id.id, type: TaxRules.TYPE };
    }
    type() {
        return TaxRules.TYPE;
    }
}
TaxRules.TYPE = 'tax_rules';
export default TaxRules;
