// @commercelayer/sdk/lib/cjs/resources/attachments.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const resource_1 = require("../resource");
class Attachments extends resource_1.ApiResource {
    async create(resource, params, options) {
        return this.resources.create({ ...resource, type: Attachments.TYPE }, params, options);
    }
    async update(resource, params, options) {
        return this.resources.update({ ...resource, type: Attachments.TYPE }, params, options);
    }
    async delete(id, options) {
        await this.resources.delete((typeof id === 'string') ? { id, type: Attachments.TYPE } : id, options);
    }
    isAttachment(resource) {
        return resource.type && (resource.type === Attachments.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: Attachments.TYPE } : { id: id.id, type: Attachments.TYPE };
    }
    type() {
        return Attachments.TYPE;
    }
}
Attachments.TYPE = 'attachments';
exports.default = Attachments;
