// @commercelayer/sdk/lib/cjs/resources/avalara_accounts.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const resource_1 = require("../resource");
class AvalaraAccounts extends resource_1.ApiResource {
    async create(resource, params, options) {
        return this.resources.create({ ...resource, type: AvalaraAccounts.TYPE }, params, options);
    }
    async update(resource, params, options) {
        return this.resources.update({ ...resource, type: AvalaraAccounts.TYPE }, params, options);
    }
    async delete(id, options) {
        await this.resources.delete((typeof id === 'string') ? { id, type: AvalaraAccounts.TYPE } : id, options);
    }
    async markets(avalaraAccountId, params, options) {
        const _avalaraAccountId = avalaraAccountId.id || avalaraAccountId;
        return this.resources.fetch({ type: 'markets' }, `avalara_accounts/${_avalaraAccountId}/markets`, params, options);
    }
    async attachments(avalaraAccountId, params, options) {
        const _avalaraAccountId = avalaraAccountId.id || avalaraAccountId;
        return this.resources.fetch({ type: 'attachments' }, `avalara_accounts/${_avalaraAccountId}/attachments`, params, options);
    }
    async versions(avalaraAccountId, params, options) {
        const _avalaraAccountId = avalaraAccountId.id || avalaraAccountId;
        return this.resources.fetch({ type: 'versions' }, `avalara_accounts/${_avalaraAccountId}/versions`, params, options);
    }
    async tax_categories(avalaraAccountId, params, options) {
        const _avalaraAccountId = avalaraAccountId.id || avalaraAccountId;
        return this.resources.fetch({ type: 'tax_categories' }, `avalara_accounts/${_avalaraAccountId}/tax_categories`, params, options);
    }
    isAvalaraAccount(resource) {
        return resource.type && (resource.type === AvalaraAccounts.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: AvalaraAccounts.TYPE } : { id: id.id, type: AvalaraAccounts.TYPE };
    }
    type() {
        return AvalaraAccounts.TYPE;
    }
}
AvalaraAccounts.TYPE = 'avalara_accounts';
exports.default = AvalaraAccounts;
