// @commercelayer/sdk/lib/cjs/resources/coupon_recipients.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const resource_1 = require("../resource");
class CouponRecipients extends resource_1.ApiResource {
    async create(resource, params, options) {
        return this.resources.create({ ...resource, type: CouponRecipients.TYPE }, params, options);
    }
    async update(resource, params, options) {
        return this.resources.update({ ...resource, type: CouponRecipients.TYPE }, params, options);
    }
    async delete(id, options) {
        await this.resources.delete((typeof id === 'string') ? { id, type: CouponRecipients.TYPE } : id, options);
    }
    async customer(couponRecipientId, params, options) {
        const _couponRecipientId = couponRecipientId.id || couponRecipientId;
        return this.resources.fetch({ type: 'customers' }, `coupon_recipients/${_couponRecipientId}/customer`, params, options);
    }
    async attachments(couponRecipientId, params, options) {
        const _couponRecipientId = couponRecipientId.id || couponRecipientId;
        return this.resources.fetch({ type: 'attachments' }, `coupon_recipients/${_couponRecipientId}/attachments`, params, options);
    }
    async versions(couponRecipientId, params, options) {
        const _couponRecipientId = couponRecipientId.id || couponRecipientId;
        return this.resources.fetch({ type: 'versions' }, `coupon_recipients/${_couponRecipientId}/versions`, params, options);
    }
    isCouponRecipient(resource) {
        return resource.type && (resource.type === CouponRecipients.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: CouponRecipients.TYPE } : { id: id.id, type: CouponRecipients.TYPE };
    }
    type() {
        return CouponRecipients.TYPE;
    }
}
CouponRecipients.TYPE = 'coupon_recipients';
exports.default = CouponRecipients;
