// @commercelayer/sdk/lib/cjs/resources/free_gift_promotions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const resource_1 = require("../resource");
class FreeGiftPromotions extends resource_1.ApiResource {
    async create(resource, params, options) {
        return this.resources.create({ ...resource, type: FreeGiftPromotions.TYPE }, params, options);
    }
    async update(resource, params, options) {
        return this.resources.update({ ...resource, type: FreeGiftPromotions.TYPE }, params, options);
    }
    async delete(id, options) {
        await this.resources.delete((typeof id === 'string') ? { id, type: FreeGiftPromotions.TYPE } : id, options);
    }
    async market(freeGiftPromotionId, params, options) {
        const _freeGiftPromotionId = freeGiftPromotionId.id || freeGiftPromotionId;
        return this.resources.fetch({ type: 'markets' }, `free_gift_promotions/${_freeGiftPromotionId}/market`, params, options);
    }
    async order_amount_promotion_rule(freeGiftPromotionId, params, options) {
        const _freeGiftPromotionId = freeGiftPromotionId.id || freeGiftPromotionId;
        return this.resources.fetch({ type: 'order_amount_promotion_rules' }, `free_gift_promotions/${_freeGiftPromotionId}/order_amount_promotion_rule`, params, options);
    }
    async sku_list_promotion_rule(freeGiftPromotionId, params, options) {
        const _freeGiftPromotionId = freeGiftPromotionId.id || freeGiftPromotionId;
        return this.resources.fetch({ type: 'sku_list_promotion_rules' }, `free_gift_promotions/${_freeGiftPromotionId}/sku_list_promotion_rule`, params, options);
    }
    async coupon_codes_promotion_rule(freeGiftPromotionId, params, options) {
        const _freeGiftPromotionId = freeGiftPromotionId.id || freeGiftPromotionId;
        return this.resources.fetch({ type: 'coupon_codes_promotion_rules' }, `free_gift_promotions/${_freeGiftPromotionId}/coupon_codes_promotion_rule`, params, options);
    }
    async coupons(freeGiftPromotionId, params, options) {
        const _freeGiftPromotionId = freeGiftPromotionId.id || freeGiftPromotionId;
        return this.resources.fetch({ type: 'coupons' }, `free_gift_promotions/${_freeGiftPromotionId}/coupons`, params, options);
    }
    async sku_list(freeGiftPromotionId, params, options) {
        const _freeGiftPromotionId = freeGiftPromotionId.id || freeGiftPromotionId;
        return this.resources.fetch({ type: 'sku_lists' }, `free_gift_promotions/${_freeGiftPromotionId}/sku_list`, params, options);
    }
    async attachments(freeGiftPromotionId, params, options) {
        const _freeGiftPromotionId = freeGiftPromotionId.id || freeGiftPromotionId;
        return this.resources.fetch({ type: 'attachments' }, `free_gift_promotions/${_freeGiftPromotionId}/attachments`, params, options);
    }
    async events(freeGiftPromotionId, params, options) {
        const _freeGiftPromotionId = freeGiftPromotionId.id || freeGiftPromotionId;
        return this.resources.fetch({ type: 'events' }, `free_gift_promotions/${_freeGiftPromotionId}/events`, params, options);
    }
    async tags(freeGiftPromotionId, params, options) {
        const _freeGiftPromotionId = freeGiftPromotionId.id || freeGiftPromotionId;
        return this.resources.fetch({ type: 'tags' }, `free_gift_promotions/${_freeGiftPromotionId}/tags`, params, options);
    }
    async versions(freeGiftPromotionId, params, options) {
        const _freeGiftPromotionId = freeGiftPromotionId.id || freeGiftPromotionId;
        return this.resources.fetch({ type: 'versions' }, `free_gift_promotions/${_freeGiftPromotionId}/versions`, params, options);
    }
    async skus(freeGiftPromotionId, params, options) {
        const _freeGiftPromotionId = freeGiftPromotionId.id || freeGiftPromotionId;
        return this.resources.fetch({ type: 'skus' }, `free_gift_promotions/${_freeGiftPromotionId}/skus`, params, options);
    }
    async _disable(id, params, options) {
        return this.resources.update({ id: (typeof id === 'string') ? id : id.id, type: FreeGiftPromotions.TYPE, _disable: true }, params, options);
    }
    async _enable(id, params, options) {
        return this.resources.update({ id: (typeof id === 'string') ? id : id.id, type: FreeGiftPromotions.TYPE, _enable: true }, params, options);
    }
    isFreeGiftPromotion(resource) {
        return resource.type && (resource.type === FreeGiftPromotions.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: FreeGiftPromotions.TYPE } : { id: id.id, type: FreeGiftPromotions.TYPE };
    }
    type() {
        return FreeGiftPromotions.TYPE;
    }
}
FreeGiftPromotions.TYPE = 'free_gift_promotions';
exports.default = FreeGiftPromotions;
