// @commercelayer/sdk/lib/cjs/resources/google_geocoders.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const resource_1 = require("../resource");
class GoogleGeocoders extends resource_1.ApiResource {
    async create(resource, params, options) {
        return this.resources.create({ ...resource, type: GoogleGeocoders.TYPE }, params, options);
    }
    async update(resource, params, options) {
        return this.resources.update({ ...resource, type: GoogleGeocoders.TYPE }, params, options);
    }
    async delete(id, options) {
        await this.resources.delete((typeof id === 'string') ? { id, type: GoogleGeocoders.TYPE } : id, options);
    }
    async addresses(googleGeocoderId, params, options) {
        const _googleGeocoderId = googleGeocoderId.id || googleGeocoderId;
        return this.resources.fetch({ type: 'addresses' }, `google_geocoders/${_googleGeocoderId}/addresses`, params, options);
    }
    async attachments(googleGeocoderId, params, options) {
        const _googleGeocoderId = googleGeocoderId.id || googleGeocoderId;
        return this.resources.fetch({ type: 'attachments' }, `google_geocoders/${_googleGeocoderId}/attachments`, params, options);
    }
    isGoogleGeocoder(resource) {
        return resource.type && (resource.type === GoogleGeocoders.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: GoogleGeocoders.TYPE } : { id: id.id, type: GoogleGeocoders.TYPE };
    }
    type() {
        return GoogleGeocoders.TYPE;
    }
}
GoogleGeocoders.TYPE = 'google_geocoders';
exports.default = GoogleGeocoders;
