// @commercelayer/sdk/lib/cjs/resources/manual_tax_calculators.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const resource_1 = require("../resource");
class ManualTaxCalculators extends resource_1.ApiResource {
    async create(resource, params, options) {
        return this.resources.create({ ...resource, type: ManualTaxCalculators.TYPE }, params, options);
    }
    async update(resource, params, options) {
        return this.resources.update({ ...resource, type: ManualTaxCalculators.TYPE }, params, options);
    }
    async delete(id, options) {
        await this.resources.delete((typeof id === 'string') ? { id, type: ManualTaxCalculators.TYPE } : id, options);
    }
    async markets(manualTaxCalculatorId, params, options) {
        const _manualTaxCalculatorId = manualTaxCalculatorId.id || manualTaxCalculatorId;
        return this.resources.fetch({ type: 'markets' }, `manual_tax_calculators/${_manualTaxCalculatorId}/markets`, params, options);
    }
    async attachments(manualTaxCalculatorId, params, options) {
        const _manualTaxCalculatorId = manualTaxCalculatorId.id || manualTaxCalculatorId;
        return this.resources.fetch({ type: 'attachments' }, `manual_tax_calculators/${_manualTaxCalculatorId}/attachments`, params, options);
    }
    async versions(manualTaxCalculatorId, params, options) {
        const _manualTaxCalculatorId = manualTaxCalculatorId.id || manualTaxCalculatorId;
        return this.resources.fetch({ type: 'versions' }, `manual_tax_calculators/${_manualTaxCalculatorId}/versions`, params, options);
    }
    async tax_rules(manualTaxCalculatorId, params, options) {
        const _manualTaxCalculatorId = manualTaxCalculatorId.id || manualTaxCalculatorId;
        return this.resources.fetch({ type: 'tax_rules' }, `manual_tax_calculators/${_manualTaxCalculatorId}/tax_rules`, params, options);
    }
    isManualTaxCalculator(resource) {
        return resource.type && (resource.type === ManualTaxCalculators.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: ManualTaxCalculators.TYPE } : { id: id.id, type: ManualTaxCalculators.TYPE };
    }
    type() {
        return ManualTaxCalculators.TYPE;
    }
}
ManualTaxCalculators.TYPE = 'manual_tax_calculators';
exports.default = ManualTaxCalculators;
