// @commercelayer/sdk/lib/cjs/resources/order_factories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const resource_1 = require("../resource");
class OrderFactories extends resource_1.ApiResource {
    async source_order(orderFactoryId, params, options) {
        const _orderFactoryId = orderFactoryId.id || orderFactoryId;
        return this.resources.fetch({ type: 'orders' }, `order_factories/${_orderFactoryId}/source_order`, params, options);
    }
    async target_order(orderFactoryId, params, options) {
        const _orderFactoryId = orderFactoryId.id || orderFactoryId;
        return this.resources.fetch({ type: 'orders' }, `order_factories/${_orderFactoryId}/target_order`, params, options);
    }
    async events(orderFactoryId, params, options) {
        const _orderFactoryId = orderFactoryId.id || orderFactoryId;
        return this.resources.fetch({ type: 'events' }, `order_factories/${_orderFactoryId}/events`, params, options);
    }
    isOrderFactory(resource) {
        return resource.type && (resource.type === OrderFactories.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: OrderFactories.TYPE } : { id: id.id, type: OrderFactories.TYPE };
    }
    type() {
        return OrderFactories.TYPE;
    }
}
OrderFactories.TYPE = 'order_factories';
exports.default = OrderFactories;
