// @commercelayer/sdk/lib/cjs/resources/payment_gateways.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const resource_1 = require("../resource");
class PaymentGateways extends resource_1.ApiResource {
    async payment_methods(paymentGatewayId, params, options) {
        const _paymentGatewayId = paymentGatewayId.id || paymentGatewayId;
        return this.resources.fetch({ type: 'payment_methods' }, `payment_gateways/${_paymentGatewayId}/payment_methods`, params, options);
    }
    async versions(paymentGatewayId, params, options) {
        const _paymentGatewayId = paymentGatewayId.id || paymentGatewayId;
        return this.resources.fetch({ type: 'versions' }, `payment_gateways/${_paymentGatewayId}/versions`, params, options);
    }
    isPaymentGateway(resource) {
        return resource.type && (resource.type === PaymentGateways.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: PaymentGateways.TYPE } : { id: id.id, type: PaymentGateways.TYPE };
    }
    type() {
        return PaymentGateways.TYPE;
    }
}
PaymentGateways.TYPE = 'payment_gateways';
exports.default = PaymentGateways;
