// @commercelayer/sdk/lib/cjs/resources/transactions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const resource_1 = require("../resource");
class Transactions extends resource_1.ApiResource {
    async order(transactionId, params, options) {
        const _transactionId = transactionId.id || transactionId;
        return this.resources.fetch({ type: 'orders' }, `transactions/${_transactionId}/order`, params, options);
    }
    async attachments(transactionId, params, options) {
        const _transactionId = transactionId.id || transactionId;
        return this.resources.fetch({ type: 'attachments' }, `transactions/${_transactionId}/attachments`, params, options);
    }
    async versions(transactionId, params, options) {
        const _transactionId = transactionId.id || transactionId;
        return this.resources.fetch({ type: 'versions' }, `transactions/${_transactionId}/versions`, params, options);
    }
    isTransaction(resource) {
        return resource.type && (resource.type === Transactions.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: Transactions.TYPE } : { id: id.id, type: Transactions.TYPE };
    }
    type() {
        return Transactions.TYPE;
    }
}
Transactions.TYPE = 'transactions';
exports.default = Transactions;
