// @commercelayer/sdk/lib/cjs/api.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.InventoryModels = exports.InStockSubscriptions = exports.Imports = exports.GoogleGeocoders = exports.GiftCards = exports.GiftCardRecipients = exports.Geocoders = exports.FreeShippingPromotions = exports.FreeGiftPromotions = exports.FixedPricePromotions = exports.FixedAmountPromotions = exports.ExternalTaxCalculators = exports.ExternalPromotions = exports.ExternalPayments = exports.ExternalGateways = exports.Exports = exports.Events = exports.EventCallbacks = exports.DeliveryLeadTimes = exports.Customers = exports.CustomerSubscriptions = exports.CustomerPaymentSources = exports.CustomerPasswordResets = exports.CustomerGroups = exports.CustomerAddresses = exports.CustomPromotionRules = exports.Coupons = exports.CouponRecipients = exports.CouponCodesPromotionRules = exports.Cleanups = exports.CheckoutComPayments = exports.CheckoutComGateways = exports.CarrierAccounts = exports.Captures = exports.BuyXPayYPromotions = exports.Bundles = exports.BraintreePayments = exports.BraintreeGateways = exports.BingGeocoders = exports.BillingInfoValidationRules = exports.AxervePayments = exports.AxerveGateways = exports.AvalaraAccounts = exports.Authorizations = exports.Attachments = exports.Applications = exports.AdyenPayments = exports.AdyenGateways = exports.Adjustments = exports.Addresses = void 0;
exports.SkuOptions = exports.SkuLists = exports.SkuListPromotionRules = exports.SkuListItems = exports.ShippingZones = exports.ShippingWeightTiers = exports.ShippingMethods = exports.ShippingMethodTiers = exports.ShippingCategories = exports.Shipments = exports.SatispayPayments = exports.SatispayGateways = exports.Returns = exports.ReturnLineItems = exports.ReservedStocks = exports.Refunds = exports.RecurringOrderCopies = exports.Promotions = exports.PromotionRules = exports.Prices = exports.PriceVolumeTiers = exports.PriceTiers = exports.PriceLists = exports.PriceFrequencyTiers = exports.PercentageDiscountPromotions = exports.PaypalPayments = exports.PaypalGateways = exports.PaymentMethods = exports.PaymentGateways = exports.Parcels = exports.ParcelLineItems = exports.Packages = exports.Organizations = exports.Orders = exports.OrderValidationRules = exports.OrderSubscriptions = exports.OrderSubscriptionItems = exports.OrderFactories = exports.OrderCopies = exports.OrderAmountPromotionRules = exports.Merchants = exports.Markets = exports.ManualTaxCalculators = exports.ManualGateways = exports.LineItems = exports.LineItemOptions = exports.KlarnaPayments = exports.KlarnaGateways = exports.InventoryStockLocations = exports.InventoryReturnLocations = void 0;
exports.resourceList = exports.WireTransfers = exports.Webhooks = exports.Voids = exports.Versions = exports.Transactions = exports.TaxjarAccounts = exports.TaxRules = exports.TaxCategories = exports.TaxCalculators = exports.Tags = exports.SubscriptionModels = exports.StripePayments = exports.StripeGateways = exports.StockTransfers = exports.StockReservations = exports.StockLocations = exports.StockLineItems = exports.StockItems = exports.Skus = void 0;
// ##__API_RESOURCES_START__##
// ##__API_RESOURCES_TEMPLATE:: export { default as ##__RESOURCE_CLASS__## } from './resources/##__RESOURCE_TYPE__##'
/**
 * ©2023 Commerce Layer Inc.
 **/
var addresses_1 = require("./resources/addresses");
Object.defineProperty(exports, "Addresses", { enumerable: true, get: function () { return __importDefault(addresses_1).default; } });
var adjustments_1 = require("./resources/adjustments");
Object.defineProperty(exports, "Adjustments", { enumerable: true, get: function () { return __importDefault(adjustments_1).default; } });
var adyen_gateways_1 = require("./resources/adyen_gateways");
Object.defineProperty(exports, "AdyenGateways", { enumerable: true, get: function () { return __importDefault(adyen_gateways_1).default; } });
var adyen_payments_1 = require("./resources/adyen_payments");
Object.defineProperty(exports, "AdyenPayments", { enumerable: true, get: function () { return __importDefault(adyen_payments_1).default; } });
var application_1 = require("./resources/application");
Object.defineProperty(exports, "Applications", { enumerable: true, get: function () { return __importDefault(application_1).default; } });
var attachments_1 = require("./resources/attachments");
Object.defineProperty(exports, "Attachments", { enumerable: true, get: function () { return __importDefault(attachments_1).default; } });
var authorizations_1 = require("./resources/authorizations");
Object.defineProperty(exports, "Authorizations", { enumerable: true, get: function () { return __importDefault(authorizations_1).default; } });
var avalara_accounts_1 = require("./resources/avalara_accounts");
Object.defineProperty(exports, "AvalaraAccounts", { enumerable: true, get: function () { return __importDefault(avalara_accounts_1).default; } });
var axerve_gateways_1 = require("./resources/axerve_gateways");
Object.defineProperty(exports, "AxerveGateways", { enumerable: true, get: function () { return __importDefault(axerve_gateways_1).default; } });
var axerve_payments_1 = require("./resources/axerve_payments");
Object.defineProperty(exports, "AxervePayments", { enumerable: true, get: function () { return __importDefault(axerve_payments_1).default; } });
var billing_info_validation_rules_1 = require("./resources/billing_info_validation_rules");
Object.defineProperty(exports, "BillingInfoValidationRules", { enumerable: true, get: function () { return __importDefault(billing_info_validation_rules_1).default; } });
var bing_geocoders_1 = require("./resources/bing_geocoders");
Object.defineProperty(exports, "BingGeocoders", { enumerable: true, get: function () { return __importDefault(bing_geocoders_1).default; } });
var braintree_gateways_1 = require("./resources/braintree_gateways");
Object.defineProperty(exports, "BraintreeGateways", { enumerable: true, get: function () { return __importDefault(braintree_gateways_1).default; } });
var braintree_payments_1 = require("./resources/braintree_payments");
Object.defineProperty(exports, "BraintreePayments", { enumerable: true, get: function () { return __importDefault(braintree_payments_1).default; } });
var bundles_1 = require("./resources/bundles");
Object.defineProperty(exports, "Bundles", { enumerable: true, get: function () { return __importDefault(bundles_1).default; } });
var buy_x_pay_y_promotions_1 = require("./resources/buy_x_pay_y_promotions");
Object.defineProperty(exports, "BuyXPayYPromotions", { enumerable: true, get: function () { return __importDefault(buy_x_pay_y_promotions_1).default; } });
var captures_1 = require("./resources/captures");
Object.defineProperty(exports, "Captures", { enumerable: true, get: function () { return __importDefault(captures_1).default; } });
var carrier_accounts_1 = require("./resources/carrier_accounts");
Object.defineProperty(exports, "CarrierAccounts", { enumerable: true, get: function () { return __importDefault(carrier_accounts_1).default; } });
var checkout_com_gateways_1 = require("./resources/checkout_com_gateways");
Object.defineProperty(exports, "CheckoutComGateways", { enumerable: true, get: function () { return __importDefault(checkout_com_gateways_1).default; } });
var checkout_com_payments_1 = require("./resources/checkout_com_payments");
Object.defineProperty(exports, "CheckoutComPayments", { enumerable: true, get: function () { return __importDefault(checkout_com_payments_1).default; } });
var cleanups_1 = require("./resources/cleanups");
Object.defineProperty(exports, "Cleanups", { enumerable: true, get: function () { return __importDefault(cleanups_1).default; } });
var coupon_codes_promotion_rules_1 = require("./resources/coupon_codes_promotion_rules");
Object.defineProperty(exports, "CouponCodesPromotionRules", { enumerable: true, get: function () { return __importDefault(coupon_codes_promotion_rules_1).default; } });
var coupon_recipients_1 = require("./resources/coupon_recipients");
Object.defineProperty(exports, "CouponRecipients", { enumerable: true, get: function () { return __importDefault(coupon_recipients_1).default; } });
var coupons_1 = require("./resources/coupons");
Object.defineProperty(exports, "Coupons", { enumerable: true, get: function () { return __importDefault(coupons_1).default; } });
var custom_promotion_rules_1 = require("./resources/custom_promotion_rules");
Object.defineProperty(exports, "CustomPromotionRules", { enumerable: true, get: function () { return __importDefault(custom_promotion_rules_1).default; } });
var customer_addresses_1 = require("./resources/customer_addresses");
Object.defineProperty(exports, "CustomerAddresses", { enumerable: true, get: function () { return __importDefault(customer_addresses_1).default; } });
var customer_groups_1 = require("./resources/customer_groups");
Object.defineProperty(exports, "CustomerGroups", { enumerable: true, get: function () { return __importDefault(customer_groups_1).default; } });
var customer_password_resets_1 = require("./resources/customer_password_resets");
Object.defineProperty(exports, "CustomerPasswordResets", { enumerable: true, get: function () { return __importDefault(customer_password_resets_1).default; } });
var customer_payment_sources_1 = require("./resources/customer_payment_sources");
Object.defineProperty(exports, "CustomerPaymentSources", { enumerable: true, get: function () { return __importDefault(customer_payment_sources_1).default; } });
var customer_subscriptions_1 = require("./resources/customer_subscriptions");
Object.defineProperty(exports, "CustomerSubscriptions", { enumerable: true, get: function () { return __importDefault(customer_subscriptions_1).default; } });
var customers_1 = require("./resources/customers");
Object.defineProperty(exports, "Customers", { enumerable: true, get: function () { return __importDefault(customers_1).default; } });
var delivery_lead_times_1 = require("./resources/delivery_lead_times");
Object.defineProperty(exports, "DeliveryLeadTimes", { enumerable: true, get: function () { return __importDefault(delivery_lead_times_1).default; } });
var event_callbacks_1 = require("./resources/event_callbacks");
Object.defineProperty(exports, "EventCallbacks", { enumerable: true, get: function () { return __importDefault(event_callbacks_1).default; } });
var events_1 = require("./resources/events");
Object.defineProperty(exports, "Events", { enumerable: true, get: function () { return __importDefault(events_1).default; } });
var exports_1 = require("./resources/exports");
Object.defineProperty(exports, "Exports", { enumerable: true, get: function () { return __importDefault(exports_1).default; } });
var external_gateways_1 = require("./resources/external_gateways");
Object.defineProperty(exports, "ExternalGateways", { enumerable: true, get: function () { return __importDefault(external_gateways_1).default; } });
var external_payments_1 = require("./resources/external_payments");
Object.defineProperty(exports, "ExternalPayments", { enumerable: true, get: function () { return __importDefault(external_payments_1).default; } });
var external_promotions_1 = require("./resources/external_promotions");
Object.defineProperty(exports, "ExternalPromotions", { enumerable: true, get: function () { return __importDefault(external_promotions_1).default; } });
var external_tax_calculators_1 = require("./resources/external_tax_calculators");
Object.defineProperty(exports, "ExternalTaxCalculators", { enumerable: true, get: function () { return __importDefault(external_tax_calculators_1).default; } });
var fixed_amount_promotions_1 = require("./resources/fixed_amount_promotions");
Object.defineProperty(exports, "FixedAmountPromotions", { enumerable: true, get: function () { return __importDefault(fixed_amount_promotions_1).default; } });
var fixed_price_promotions_1 = require("./resources/fixed_price_promotions");
Object.defineProperty(exports, "FixedPricePromotions", { enumerable: true, get: function () { return __importDefault(fixed_price_promotions_1).default; } });
var free_gift_promotions_1 = require("./resources/free_gift_promotions");
Object.defineProperty(exports, "FreeGiftPromotions", { enumerable: true, get: function () { return __importDefault(free_gift_promotions_1).default; } });
var free_shipping_promotions_1 = require("./resources/free_shipping_promotions");
Object.defineProperty(exports, "FreeShippingPromotions", { enumerable: true, get: function () { return __importDefault(free_shipping_promotions_1).default; } });
var geocoders_1 = require("./resources/geocoders");
Object.defineProperty(exports, "Geocoders", { enumerable: true, get: function () { return __importDefault(geocoders_1).default; } });
var gift_card_recipients_1 = require("./resources/gift_card_recipients");
Object.defineProperty(exports, "GiftCardRecipients", { enumerable: true, get: function () { return __importDefault(gift_card_recipients_1).default; } });
var gift_cards_1 = require("./resources/gift_cards");
Object.defineProperty(exports, "GiftCards", { enumerable: true, get: function () { return __importDefault(gift_cards_1).default; } });
var google_geocoders_1 = require("./resources/google_geocoders");
Object.defineProperty(exports, "GoogleGeocoders", { enumerable: true, get: function () { return __importDefault(google_geocoders_1).default; } });
var imports_1 = require("./resources/imports");
Object.defineProperty(exports, "Imports", { enumerable: true, get: function () { return __importDefault(imports_1).default; } });
var in_stock_subscriptions_1 = require("./resources/in_stock_subscriptions");
Object.defineProperty(exports, "InStockSubscriptions", { enumerable: true, get: function () { return __importDefault(in_stock_subscriptions_1).default; } });
var inventory_models_1 = require("./resources/inventory_models");
Object.defineProperty(exports, "InventoryModels", { enumerable: true, get: function () { return __importDefault(inventory_models_1).default; } });
var inventory_return_locations_1 = require("./resources/inventory_return_locations");
Object.defineProperty(exports, "InventoryReturnLocations", { enumerable: true, get: function () { return __importDefault(inventory_return_locations_1).default; } });
var inventory_stock_locations_1 = require("./resources/inventory_stock_locations");
Object.defineProperty(exports, "InventoryStockLocations", { enumerable: true, get: function () { return __importDefault(inventory_stock_locations_1).default; } });
var klarna_gateways_1 = require("./resources/klarna_gateways");
Object.defineProperty(exports, "KlarnaGateways", { enumerable: true, get: function () { return __importDefault(klarna_gateways_1).default; } });
var klarna_payments_1 = require("./resources/klarna_payments");
Object.defineProperty(exports, "KlarnaPayments", { enumerable: true, get: function () { return __importDefault(klarna_payments_1).default; } });
var line_item_options_1 = require("./resources/line_item_options");
Object.defineProperty(exports, "LineItemOptions", { enumerable: true, get: function () { return __importDefault(line_item_options_1).default; } });
var line_items_1 = require("./resources/line_items");
Object.defineProperty(exports, "LineItems", { enumerable: true, get: function () { return __importDefault(line_items_1).default; } });
var manual_gateways_1 = require("./resources/manual_gateways");
Object.defineProperty(exports, "ManualGateways", { enumerable: true, get: function () { return __importDefault(manual_gateways_1).default; } });
var manual_tax_calculators_1 = require("./resources/manual_tax_calculators");
Object.defineProperty(exports, "ManualTaxCalculators", { enumerable: true, get: function () { return __importDefault(manual_tax_calculators_1).default; } });
var markets_1 = require("./resources/markets");
Object.defineProperty(exports, "Markets", { enumerable: true, get: function () { return __importDefault(markets_1).default; } });
var merchants_1 = require("./resources/merchants");
Object.defineProperty(exports, "Merchants", { enumerable: true, get: function () { return __importDefault(merchants_1).default; } });
var order_amount_promotion_rules_1 = require("./resources/order_amount_promotion_rules");
Object.defineProperty(exports, "OrderAmountPromotionRules", { enumerable: true, get: function () { return __importDefault(order_amount_promotion_rules_1).default; } });
var order_copies_1 = require("./resources/order_copies");
Object.defineProperty(exports, "OrderCopies", { enumerable: true, get: function () { return __importDefault(order_copies_1).default; } });
var order_factories_1 = require("./resources/order_factories");
Object.defineProperty(exports, "OrderFactories", { enumerable: true, get: function () { return __importDefault(order_factories_1).default; } });
var order_subscription_items_1 = require("./resources/order_subscription_items");
Object.defineProperty(exports, "OrderSubscriptionItems", { enumerable: true, get: function () { return __importDefault(order_subscription_items_1).default; } });
var order_subscriptions_1 = require("./resources/order_subscriptions");
Object.defineProperty(exports, "OrderSubscriptions", { enumerable: true, get: function () { return __importDefault(order_subscriptions_1).default; } });
var order_validation_rules_1 = require("./resources/order_validation_rules");
Object.defineProperty(exports, "OrderValidationRules", { enumerable: true, get: function () { return __importDefault(order_validation_rules_1).default; } });
var orders_1 = require("./resources/orders");
Object.defineProperty(exports, "Orders", { enumerable: true, get: function () { return __importDefault(orders_1).default; } });
var organization_1 = require("./resources/organization");
Object.defineProperty(exports, "Organizations", { enumerable: true, get: function () { return __importDefault(organization_1).default; } });
var packages_1 = require("./resources/packages");
Object.defineProperty(exports, "Packages", { enumerable: true, get: function () { return __importDefault(packages_1).default; } });
var parcel_line_items_1 = require("./resources/parcel_line_items");
Object.defineProperty(exports, "ParcelLineItems", { enumerable: true, get: function () { return __importDefault(parcel_line_items_1).default; } });
var parcels_1 = require("./resources/parcels");
Object.defineProperty(exports, "Parcels", { enumerable: true, get: function () { return __importDefault(parcels_1).default; } });
var payment_gateways_1 = require("./resources/payment_gateways");
Object.defineProperty(exports, "PaymentGateways", { enumerable: true, get: function () { return __importDefault(payment_gateways_1).default; } });
var payment_methods_1 = require("./resources/payment_methods");
Object.defineProperty(exports, "PaymentMethods", { enumerable: true, get: function () { return __importDefault(payment_methods_1).default; } });
var paypal_gateways_1 = require("./resources/paypal_gateways");
Object.defineProperty(exports, "PaypalGateways", { enumerable: true, get: function () { return __importDefault(paypal_gateways_1).default; } });
var paypal_payments_1 = require("./resources/paypal_payments");
Object.defineProperty(exports, "PaypalPayments", { enumerable: true, get: function () { return __importDefault(paypal_payments_1).default; } });
var percentage_discount_promotions_1 = require("./resources/percentage_discount_promotions");
Object.defineProperty(exports, "PercentageDiscountPromotions", { enumerable: true, get: function () { return __importDefault(percentage_discount_promotions_1).default; } });
var price_frequency_tiers_1 = require("./resources/price_frequency_tiers");
Object.defineProperty(exports, "PriceFrequencyTiers", { enumerable: true, get: function () { return __importDefault(price_frequency_tiers_1).default; } });
var price_lists_1 = require("./resources/price_lists");
Object.defineProperty(exports, "PriceLists", { enumerable: true, get: function () { return __importDefault(price_lists_1).default; } });
var price_tiers_1 = require("./resources/price_tiers");
Object.defineProperty(exports, "PriceTiers", { enumerable: true, get: function () { return __importDefault(price_tiers_1).default; } });
var price_volume_tiers_1 = require("./resources/price_volume_tiers");
Object.defineProperty(exports, "PriceVolumeTiers", { enumerable: true, get: function () { return __importDefault(price_volume_tiers_1).default; } });
var prices_1 = require("./resources/prices");
Object.defineProperty(exports, "Prices", { enumerable: true, get: function () { return __importDefault(prices_1).default; } });
var promotion_rules_1 = require("./resources/promotion_rules");
Object.defineProperty(exports, "PromotionRules", { enumerable: true, get: function () { return __importDefault(promotion_rules_1).default; } });
var promotions_1 = require("./resources/promotions");
Object.defineProperty(exports, "Promotions", { enumerable: true, get: function () { return __importDefault(promotions_1).default; } });
var recurring_order_copies_1 = require("./resources/recurring_order_copies");
Object.defineProperty(exports, "RecurringOrderCopies", { enumerable: true, get: function () { return __importDefault(recurring_order_copies_1).default; } });
var refunds_1 = require("./resources/refunds");
Object.defineProperty(exports, "Refunds", { enumerable: true, get: function () { return __importDefault(refunds_1).default; } });
var reserved_stocks_1 = require("./resources/reserved_stocks");
Object.defineProperty(exports, "ReservedStocks", { enumerable: true, get: function () { return __importDefault(reserved_stocks_1).default; } });
var return_line_items_1 = require("./resources/return_line_items");
Object.defineProperty(exports, "ReturnLineItems", { enumerable: true, get: function () { return __importDefault(return_line_items_1).default; } });
var returns_1 = require("./resources/returns");
Object.defineProperty(exports, "Returns", { enumerable: true, get: function () { return __importDefault(returns_1).default; } });
var satispay_gateways_1 = require("./resources/satispay_gateways");
Object.defineProperty(exports, "SatispayGateways", { enumerable: true, get: function () { return __importDefault(satispay_gateways_1).default; } });
var satispay_payments_1 = require("./resources/satispay_payments");
Object.defineProperty(exports, "SatispayPayments", { enumerable: true, get: function () { return __importDefault(satispay_payments_1).default; } });
var shipments_1 = require("./resources/shipments");
Object.defineProperty(exports, "Shipments", { enumerable: true, get: function () { return __importDefault(shipments_1).default; } });
var shipping_categories_1 = require("./resources/shipping_categories");
Object.defineProperty(exports, "ShippingCategories", { enumerable: true, get: function () { return __importDefault(shipping_categories_1).default; } });
var shipping_method_tiers_1 = require("./resources/shipping_method_tiers");
Object.defineProperty(exports, "ShippingMethodTiers", { enumerable: true, get: function () { return __importDefault(shipping_method_tiers_1).default; } });
var shipping_methods_1 = require("./resources/shipping_methods");
Object.defineProperty(exports, "ShippingMethods", { enumerable: true, get: function () { return __importDefault(shipping_methods_1).default; } });
var shipping_weight_tiers_1 = require("./resources/shipping_weight_tiers");
Object.defineProperty(exports, "ShippingWeightTiers", { enumerable: true, get: function () { return __importDefault(shipping_weight_tiers_1).default; } });
var shipping_zones_1 = require("./resources/shipping_zones");
Object.defineProperty(exports, "ShippingZones", { enumerable: true, get: function () { return __importDefault(shipping_zones_1).default; } });
var sku_list_items_1 = require("./resources/sku_list_items");
Object.defineProperty(exports, "SkuListItems", { enumerable: true, get: function () { return __importDefault(sku_list_items_1).default; } });
var sku_list_promotion_rules_1 = require("./resources/sku_list_promotion_rules");
Object.defineProperty(exports, "SkuListPromotionRules", { enumerable: true, get: function () { return __importDefault(sku_list_promotion_rules_1).default; } });
var sku_lists_1 = require("./resources/sku_lists");
Object.defineProperty(exports, "SkuLists", { enumerable: true, get: function () { return __importDefault(sku_lists_1).default; } });
var sku_options_1 = require("./resources/sku_options");
Object.defineProperty(exports, "SkuOptions", { enumerable: true, get: function () { return __importDefault(sku_options_1).default; } });
var skus_1 = require("./resources/skus");
Object.defineProperty(exports, "Skus", { enumerable: true, get: function () { return __importDefault(skus_1).default; } });
var stock_items_1 = require("./resources/stock_items");
Object.defineProperty(exports, "StockItems", { enumerable: true, get: function () { return __importDefault(stock_items_1).default; } });
var stock_line_items_1 = require("./resources/stock_line_items");
Object.defineProperty(exports, "StockLineItems", { enumerable: true, get: function () { return __importDefault(stock_line_items_1).default; } });
var stock_locations_1 = require("./resources/stock_locations");
Object.defineProperty(exports, "StockLocations", { enumerable: true, get: function () { return __importDefault(stock_locations_1).default; } });
var stock_reservations_1 = require("./resources/stock_reservations");
Object.defineProperty(exports, "StockReservations", { enumerable: true, get: function () { return __importDefault(stock_reservations_1).default; } });
var stock_transfers_1 = require("./resources/stock_transfers");
Object.defineProperty(exports, "StockTransfers", { enumerable: true, get: function () { return __importDefault(stock_transfers_1).default; } });
var stripe_gateways_1 = require("./resources/stripe_gateways");
Object.defineProperty(exports, "StripeGateways", { enumerable: true, get: function () { return __importDefault(stripe_gateways_1).default; } });
var stripe_payments_1 = require("./resources/stripe_payments");
Object.defineProperty(exports, "StripePayments", { enumerable: true, get: function () { return __importDefault(stripe_payments_1).default; } });
var subscription_models_1 = require("./resources/subscription_models");
Object.defineProperty(exports, "SubscriptionModels", { enumerable: true, get: function () { return __importDefault(subscription_models_1).default; } });
var tags_1 = require("./resources/tags");
Object.defineProperty(exports, "Tags", { enumerable: true, get: function () { return __importDefault(tags_1).default; } });
var tax_calculators_1 = require("./resources/tax_calculators");
Object.defineProperty(exports, "TaxCalculators", { enumerable: true, get: function () { return __importDefault(tax_calculators_1).default; } });
var tax_categories_1 = require("./resources/tax_categories");
Object.defineProperty(exports, "TaxCategories", { enumerable: true, get: function () { return __importDefault(tax_categories_1).default; } });
var tax_rules_1 = require("./resources/tax_rules");
Object.defineProperty(exports, "TaxRules", { enumerable: true, get: function () { return __importDefault(tax_rules_1).default; } });
var taxjar_accounts_1 = require("./resources/taxjar_accounts");
Object.defineProperty(exports, "TaxjarAccounts", { enumerable: true, get: function () { return __importDefault(taxjar_accounts_1).default; } });
var transactions_1 = require("./resources/transactions");
Object.defineProperty(exports, "Transactions", { enumerable: true, get: function () { return __importDefault(transactions_1).default; } });
var versions_1 = require("./resources/versions");
Object.defineProperty(exports, "Versions", { enumerable: true, get: function () { return __importDefault(versions_1).default; } });
var voids_1 = require("./resources/voids");
Object.defineProperty(exports, "Voids", { enumerable: true, get: function () { return __importDefault(voids_1).default; } });
var webhooks_1 = require("./resources/webhooks");
Object.defineProperty(exports, "Webhooks", { enumerable: true, get: function () { return __importDefault(webhooks_1).default; } });
var wire_transfers_1 = require("./resources/wire_transfers");
Object.defineProperty(exports, "WireTransfers", { enumerable: true, get: function () { return __importDefault(wire_transfers_1).default; } });
// ##__API_RESOURCE_TYPES_STOP__##
exports.resourceList = [
    // ##__API_RESOURCE_LIST_START__##
    'addresses',
    'adjustments',
    'adyen_gateways',
    'adyen_payments',
    'application',
    'attachments',
    'authorizations',
    'avalara_accounts',
    'axerve_gateways',
    'axerve_payments',
    'billing_info_validation_rules',
    'bing_geocoders',
    'braintree_gateways',
    'braintree_payments',
    'bundles',
    'buy_x_pay_y_promotions',
    'captures',
    'carrier_accounts',
    'checkout_com_gateways',
    'checkout_com_payments',
    'cleanups',
    'coupon_codes_promotion_rules',
    'coupon_recipients',
    'coupons',
    'custom_promotion_rules',
    'customer_addresses',
    'customer_groups',
    'customer_password_resets',
    'customer_payment_sources',
    'customer_subscriptions',
    'customers',
    'delivery_lead_times',
    'event_callbacks',
    'events',
    'exports',
    'external_gateways',
    'external_payments',
    'external_promotions',
    'external_tax_calculators',
    'fixed_amount_promotions',
    'fixed_price_promotions',
    'free_gift_promotions',
    'free_shipping_promotions',
    'geocoders',
    'gift_card_recipients',
    'gift_cards',
    'google_geocoders',
    'imports',
    'in_stock_subscriptions',
    'inventory_models',
    'inventory_return_locations',
    'inventory_stock_locations',
    'klarna_gateways',
    'klarna_payments',
    'line_item_options',
    'line_items',
    'manual_gateways',
    'manual_tax_calculators',
    'markets',
    'merchants',
    'order_amount_promotion_rules',
    'order_copies',
    'order_factories',
    'order_subscription_items',
    'order_subscriptions',
    'order_validation_rules',
    'orders',
    'organization',
    'packages',
    'parcel_line_items',
    'parcels',
    'payment_gateways',
    'payment_methods',
    'paypal_gateways',
    'paypal_payments',
    'percentage_discount_promotions',
    'price_frequency_tiers',
    'price_lists',
    'price_tiers',
    'price_volume_tiers',
    'prices',
    'promotion_rules',
    'promotions',
    'recurring_order_copies',
    'refunds',
    'reserved_stocks',
    'return_line_items',
    'returns',
    'satispay_gateways',
    'satispay_payments',
    'shipments',
    'shipping_categories',
    'shipping_method_tiers',
    'shipping_methods',
    'shipping_weight_tiers',
    'shipping_zones',
    'sku_list_items',
    'sku_list_promotion_rules',
    'sku_lists',
    'sku_options',
    'skus',
    'stock_items',
    'stock_line_items',
    'stock_locations',
    'stock_reservations',
    'stock_transfers',
    'stripe_gateways',
    'stripe_payments',
    'subscription_models',
    'tags',
    'tax_calculators',
    'tax_categories',
    'tax_rules',
    'taxjar_accounts',
    'transactions',
    'versions',
    'voids',
    'webhooks',
    'wire_transfers'
    // ##__API_RESOURCE_LIST_STOP__##
];
