// rapid-form/lib/cjs/hooks/useRapidForm.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.useRapidForm = void 0;
const tslib_1 = require("tslib");
const react_1 = require("react");
const fetchReducer_1 = tslib_1.__importDefault(require("../utils/fetchReducer"));
const useValidation_1 = tslib_1.__importDefault(require("./useValidation"));
const reset_1 = tslib_1.__importDefault(require("../utils/reset"));
const reset_2 = require("../utils/reset");
const useSubmitValidation_1 = tslib_1.__importDefault(require("./useSubmitValidation "));
const isEmpty_1 = tslib_1.__importDefault(require("lodash/isEmpty"));
const map_1 = tslib_1.__importDefault(require("lodash/map"));
const handleChange_1 = tslib_1.__importDefault(require("../utils/handleChange"));
const setErrors_1 = tslib_1.__importDefault(require("../utils/setErrors"));
function useRapidForm(config) {
    const [state, dispatch] = (0, react_1.useReducer)(fetchReducer_1.default, {
        data: {},
        errors: {},
        refs: {},
    });
    const ValidationHook = (0, useValidation_1.default)(dispatch, config?.fieldEvent);
    const SubmitValidation = (0, useSubmitValidation_1.default)(dispatch);
    const reset = (e, name) => {
        const target = e.currentTarget || e.target;
        if (name) {
            const field = target.elements.namedItem(name);
            if (field.type === 'checkbox' || field.type === 'radio')
                field.checked = false;
            field.value = '';
            (0, reset_2.resetSingleField)(field, dispatch);
        }
        else {
            target.reset();
            (0, reset_1.default)(dispatch);
        }
    };
    return {
        handleSubmit: (c) => (e) => {
            let tempState = {
                data: {},
                errors: {},
            };
            e.preventDefault();
            (0, map_1.default)(e.currentTarget.elements, (e) => {
                if (e.name) {
                    const st = (0, handleChange_1.default)(e, dispatch);
                    tempState = {
                        data: { ...tempState.data, ...st.data },
                        errors: { ...tempState.errors, ...st.errors },
                    };
                }
            });
            const newState = {
                data: (0, isEmpty_1.default)(state.data) ? tempState.data : state.data,
                errors: (0, isEmpty_1.default)(state.errors) ? tempState.errors : state.errors,
            };
            return c(newState.data, newState.errors, e);
        },
        errors: state.errors,
        validation: ValidationHook,
        submitValidation: SubmitValidation,
        reset,
        values: state.data,
        setValue: (name, value) => {
            const field = state.refs[name];
            if (field) {
                field.value = value;
                (0, setErrors_1.default)({
                    checked: field.checked,
                    name,
                    pattern: field.pattern,
                    required: field.required,
                    type: field.type,
                    value,
                }, dispatch);
            }
        },
        setError: (error) => {
            const { name, ...e } = error;
            dispatch({
                type: 'error',
                name,
                errors: {
                    [name]: {
                        ...e,
                        error: true,
                    },
                },
            });
        },
    };
}
exports.useRapidForm = useRapidForm;
