// @commercelayer/sdk/lib/esm/resources/adyen_gateways.js
import { ApiResource } from '../resource';
class AdyenGateways extends ApiResource {
    async create(resource, params, options) {
        return this.resources.create({ ...resource, type: AdyenGateways.TYPE }, params, options);
    }
    async update(resource, params, options) {
        return this.resources.update({ ...resource, type: AdyenGateways.TYPE }, params, options);
    }
    async delete(id, options) {
        await this.resources.delete((typeof id === 'string') ? { id, type: AdyenGateways.TYPE } : id, options);
    }
    async payment_methods(adyenGatewayId, params, options) {
        const _adyenGatewayId = adyenGatewayId.id || adyenGatewayId;
        return this.resources.fetch({ type: 'payment_methods' }, `adyen_gateways/${_adyenGatewayId}/payment_methods`, params, options);
    }
    async versions(adyenGatewayId, params, options) {
        const _adyenGatewayId = adyenGatewayId.id || adyenGatewayId;
        return this.resources.fetch({ type: 'versions' }, `adyen_gateways/${_adyenGatewayId}/versions`, params, options);
    }
    async adyen_payments(adyenGatewayId, params, options) {
        const _adyenGatewayId = adyenGatewayId.id || adyenGatewayId;
        return this.resources.fetch({ type: 'adyen_payments' }, `adyen_gateways/${_adyenGatewayId}/adyen_payments`, params, options);
    }
    isAdyenGateway(resource) {
        return resource.type && (resource.type === AdyenGateways.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: AdyenGateways.TYPE } : { id: id.id, type: AdyenGateways.TYPE };
    }
    type() {
        return AdyenGateways.TYPE;
    }
}
AdyenGateways.TYPE = 'adyen_gateways';
export default AdyenGateways;
