// @commercelayer/sdk/lib/esm/resources/billing_info_validation_rules.js
import { ApiResource } from '../resource';
class BillingInfoValidationRules extends ApiResource {
    async create(resource, params, options) {
        return this.resources.create({ ...resource, type: BillingInfoValidationRules.TYPE }, params, options);
    }
    async update(resource, params, options) {
        return this.resources.update({ ...resource, type: BillingInfoValidationRules.TYPE }, params, options);
    }
    async delete(id, options) {
        await this.resources.delete((typeof id === 'string') ? { id, type: BillingInfoValidationRules.TYPE } : id, options);
    }
    async market(billingInfoValidationRuleId, params, options) {
        const _billingInfoValidationRuleId = billingInfoValidationRuleId.id || billingInfoValidationRuleId;
        return this.resources.fetch({ type: 'markets' }, `billing_info_validation_rules/${_billingInfoValidationRuleId}/market`, params, options);
    }
    async attachments(billingInfoValidationRuleId, params, options) {
        const _billingInfoValidationRuleId = billingInfoValidationRuleId.id || billingInfoValidationRuleId;
        return this.resources.fetch({ type: 'attachments' }, `billing_info_validation_rules/${_billingInfoValidationRuleId}/attachments`, params, options);
    }
    async versions(billingInfoValidationRuleId, params, options) {
        const _billingInfoValidationRuleId = billingInfoValidationRuleId.id || billingInfoValidationRuleId;
        return this.resources.fetch({ type: 'versions' }, `billing_info_validation_rules/${_billingInfoValidationRuleId}/versions`, params, options);
    }
    isBillingInfoValidationRule(resource) {
        return resource.type && (resource.type === BillingInfoValidationRules.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: BillingInfoValidationRules.TYPE } : { id: id.id, type: BillingInfoValidationRules.TYPE };
    }
    type() {
        return BillingInfoValidationRules.TYPE;
    }
}
BillingInfoValidationRules.TYPE = 'billing_info_validation_rules';
export default BillingInfoValidationRules;
