// @commercelayer/sdk/lib/esm/resources/coupon_codes_promotion_rules.js
import { ApiResource } from '../resource';
class CouponCodesPromotionRules extends ApiResource {
    async create(resource, params, options) {
        return this.resources.create({ ...resource, type: CouponCodesPromotionRules.TYPE }, params, options);
    }
    async update(resource, params, options) {
        return this.resources.update({ ...resource, type: CouponCodesPromotionRules.TYPE }, params, options);
    }
    async delete(id, options) {
        await this.resources.delete((typeof id === 'string') ? { id, type: CouponCodesPromotionRules.TYPE } : id, options);
    }
    async versions(couponCodesPromotionRuleId, params, options) {
        const _couponCodesPromotionRuleId = couponCodesPromotionRuleId.id || couponCodesPromotionRuleId;
        return this.resources.fetch({ type: 'versions' }, `coupon_codes_promotion_rules/${_couponCodesPromotionRuleId}/versions`, params, options);
    }
    async coupons(couponCodesPromotionRuleId, params, options) {
        const _couponCodesPromotionRuleId = couponCodesPromotionRuleId.id || couponCodesPromotionRuleId;
        return this.resources.fetch({ type: 'coupons' }, `coupon_codes_promotion_rules/${_couponCodesPromotionRuleId}/coupons`, params, options);
    }
    isCouponCodesPromotionRule(resource) {
        return resource.type && (resource.type === CouponCodesPromotionRules.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: CouponCodesPromotionRules.TYPE } : { id: id.id, type: CouponCodesPromotionRules.TYPE };
    }
    type() {
        return CouponCodesPromotionRules.TYPE;
    }
}
CouponCodesPromotionRules.TYPE = 'coupon_codes_promotion_rules';
export default CouponCodesPromotionRules;
