// @commercelayer/sdk/lib/esm/resources/external_promotions.js
import { ApiResource } from '../resource';
class ExternalPromotions extends ApiResource {
    async create(resource, params, options) {
        return this.resources.create({ ...resource, type: ExternalPromotions.TYPE }, params, options);
    }
    async update(resource, params, options) {
        return this.resources.update({ ...resource, type: ExternalPromotions.TYPE }, params, options);
    }
    async delete(id, options) {
        await this.resources.delete((typeof id === 'string') ? { id, type: ExternalPromotions.TYPE } : id, options);
    }
    async market(externalPromotionId, params, options) {
        const _externalPromotionId = externalPromotionId.id || externalPromotionId;
        return this.resources.fetch({ type: 'markets' }, `external_promotions/${_externalPromotionId}/market`, params, options);
    }
    async order_amount_promotion_rule(externalPromotionId, params, options) {
        const _externalPromotionId = externalPromotionId.id || externalPromotionId;
        return this.resources.fetch({ type: 'order_amount_promotion_rules' }, `external_promotions/${_externalPromotionId}/order_amount_promotion_rule`, params, options);
    }
    async sku_list_promotion_rule(externalPromotionId, params, options) {
        const _externalPromotionId = externalPromotionId.id || externalPromotionId;
        return this.resources.fetch({ type: 'sku_list_promotion_rules' }, `external_promotions/${_externalPromotionId}/sku_list_promotion_rule`, params, options);
    }
    async coupon_codes_promotion_rule(externalPromotionId, params, options) {
        const _externalPromotionId = externalPromotionId.id || externalPromotionId;
        return this.resources.fetch({ type: 'coupon_codes_promotion_rules' }, `external_promotions/${_externalPromotionId}/coupon_codes_promotion_rule`, params, options);
    }
    async coupons(externalPromotionId, params, options) {
        const _externalPromotionId = externalPromotionId.id || externalPromotionId;
        return this.resources.fetch({ type: 'coupons' }, `external_promotions/${_externalPromotionId}/coupons`, params, options);
    }
    async sku_list(externalPromotionId, params, options) {
        const _externalPromotionId = externalPromotionId.id || externalPromotionId;
        return this.resources.fetch({ type: 'sku_lists' }, `external_promotions/${_externalPromotionId}/sku_list`, params, options);
    }
    async attachments(externalPromotionId, params, options) {
        const _externalPromotionId = externalPromotionId.id || externalPromotionId;
        return this.resources.fetch({ type: 'attachments' }, `external_promotions/${_externalPromotionId}/attachments`, params, options);
    }
    async events(externalPromotionId, params, options) {
        const _externalPromotionId = externalPromotionId.id || externalPromotionId;
        return this.resources.fetch({ type: 'events' }, `external_promotions/${_externalPromotionId}/events`, params, options);
    }
    async tags(externalPromotionId, params, options) {
        const _externalPromotionId = externalPromotionId.id || externalPromotionId;
        return this.resources.fetch({ type: 'tags' }, `external_promotions/${_externalPromotionId}/tags`, params, options);
    }
    async versions(externalPromotionId, params, options) {
        const _externalPromotionId = externalPromotionId.id || externalPromotionId;
        return this.resources.fetch({ type: 'versions' }, `external_promotions/${_externalPromotionId}/versions`, params, options);
    }
    async skus(externalPromotionId, params, options) {
        const _externalPromotionId = externalPromotionId.id || externalPromotionId;
        return this.resources.fetch({ type: 'skus' }, `external_promotions/${_externalPromotionId}/skus`, params, options);
    }
    async _disable(id, params, options) {
        return this.resources.update({ id: (typeof id === 'string') ? id : id.id, type: ExternalPromotions.TYPE, _disable: true }, params, options);
    }
    async _enable(id, params, options) {
        return this.resources.update({ id: (typeof id === 'string') ? id : id.id, type: ExternalPromotions.TYPE, _enable: true }, params, options);
    }
    isExternalPromotion(resource) {
        return resource.type && (resource.type === ExternalPromotions.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: ExternalPromotions.TYPE } : { id: id.id, type: ExternalPromotions.TYPE };
    }
    type() {
        return ExternalPromotions.TYPE;
    }
}
ExternalPromotions.TYPE = 'external_promotions';
export default ExternalPromotions;
