// @commercelayer/sdk/lib/esm/resources/line_item_options.js
import { ApiResource } from '../resource';
class LineItemOptions extends ApiResource {
    async create(resource, params, options) {
        return this.resources.create({ ...resource, type: LineItemOptions.TYPE }, params, options);
    }
    async update(resource, params, options) {
        return this.resources.update({ ...resource, type: LineItemOptions.TYPE }, params, options);
    }
    async delete(id, options) {
        await this.resources.delete((typeof id === 'string') ? { id, type: LineItemOptions.TYPE } : id, options);
    }
    async line_item(lineItemOptionId, params, options) {
        const _lineItemOptionId = lineItemOptionId.id || lineItemOptionId;
        return this.resources.fetch({ type: 'line_items' }, `line_item_options/${_lineItemOptionId}/line_item`, params, options);
    }
    async sku_option(lineItemOptionId, params, options) {
        const _lineItemOptionId = lineItemOptionId.id || lineItemOptionId;
        return this.resources.fetch({ type: 'sku_options' }, `line_item_options/${_lineItemOptionId}/sku_option`, params, options);
    }
    async events(lineItemOptionId, params, options) {
        const _lineItemOptionId = lineItemOptionId.id || lineItemOptionId;
        return this.resources.fetch({ type: 'events' }, `line_item_options/${_lineItemOptionId}/events`, params, options);
    }
    async tags(lineItemOptionId, params, options) {
        const _lineItemOptionId = lineItemOptionId.id || lineItemOptionId;
        return this.resources.fetch({ type: 'tags' }, `line_item_options/${_lineItemOptionId}/tags`, params, options);
    }
    isLineItemOption(resource) {
        return resource.type && (resource.type === LineItemOptions.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: LineItemOptions.TYPE } : { id: id.id, type: LineItemOptions.TYPE };
    }
    type() {
        return LineItemOptions.TYPE;
    }
}
LineItemOptions.TYPE = 'line_item_options';
export default LineItemOptions;
