// @commercelayer/sdk/lib/esm/resources/order_subscription_items.js
import { ApiResource } from '../resource';
class OrderSubscriptionItems extends ApiResource {
    async create(resource, params, options) {
        return this.resources.create({ ...resource, type: OrderSubscriptionItems.TYPE }, params, options);
    }
    async update(resource, params, options) {
        return this.resources.update({ ...resource, type: OrderSubscriptionItems.TYPE }, params, options);
    }
    async delete(id, options) {
        await this.resources.delete((typeof id === 'string') ? { id, type: OrderSubscriptionItems.TYPE } : id, options);
    }
    async order_subscription(orderSubscriptionItemId, params, options) {
        const _orderSubscriptionItemId = orderSubscriptionItemId.id || orderSubscriptionItemId;
        return this.resources.fetch({ type: 'order_subscriptions' }, `order_subscription_items/${_orderSubscriptionItemId}/order_subscription`, params, options);
    }
    async source_line_item(orderSubscriptionItemId, params, options) {
        const _orderSubscriptionItemId = orderSubscriptionItemId.id || orderSubscriptionItemId;
        return this.resources.fetch({ type: 'line_items' }, `order_subscription_items/${_orderSubscriptionItemId}/source_line_item`, params, options);
    }
    isOrderSubscriptionItem(resource) {
        return resource.type && (resource.type === OrderSubscriptionItems.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: OrderSubscriptionItems.TYPE } : { id: id.id, type: OrderSubscriptionItems.TYPE };
    }
    type() {
        return OrderSubscriptionItems.TYPE;
    }
}
OrderSubscriptionItems.TYPE = 'order_subscription_items';
export default OrderSubscriptionItems;
