// @commercelayer/sdk/lib/esm/resources/paypal_payments.js
import { ApiResource } from '../resource';
class PaypalPayments extends ApiResource {
    async create(resource, params, options) {
        return this.resources.create({ ...resource, type: PaypalPayments.TYPE }, params, options);
    }
    async update(resource, params, options) {
        return this.resources.update({ ...resource, type: PaypalPayments.TYPE }, params, options);
    }
    async delete(id, options) {
        await this.resources.delete((typeof id === 'string') ? { id, type: PaypalPayments.TYPE } : id, options);
    }
    async order(paypalPaymentId, params, options) {
        const _paypalPaymentId = paypalPaymentId.id || paypalPaymentId;
        return this.resources.fetch({ type: 'orders' }, `paypal_payments/${_paypalPaymentId}/order`, params, options);
    }
    async payment_gateway(paypalPaymentId, params, options) {
        const _paypalPaymentId = paypalPaymentId.id || paypalPaymentId;
        return this.resources.fetch({ type: 'payment_gateways' }, `paypal_payments/${_paypalPaymentId}/payment_gateway`, params, options);
    }
    async versions(paypalPaymentId, params, options) {
        const _paypalPaymentId = paypalPaymentId.id || paypalPaymentId;
        return this.resources.fetch({ type: 'versions' }, `paypal_payments/${_paypalPaymentId}/versions`, params, options);
    }
    isPaypalPayment(resource) {
        return resource.type && (resource.type === PaypalPayments.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: PaypalPayments.TYPE } : { id: id.id, type: PaypalPayments.TYPE };
    }
    type() {
        return PaypalPayments.TYPE;
    }
}
PaypalPayments.TYPE = 'paypal_payments';
export default PaypalPayments;
