// @commercelayer/sdk/lib/esm/resources/reserved_stocks.js
import { ApiResource } from '../resource';
class ReservedStocks extends ApiResource {
    async stock_item(reservedStockId, params, options) {
        const _reservedStockId = reservedStockId.id || reservedStockId;
        return this.resources.fetch({ type: 'stock_items' }, `reserved_stocks/${_reservedStockId}/stock_item`, params, options);
    }
    async sku(reservedStockId, params, options) {
        const _reservedStockId = reservedStockId.id || reservedStockId;
        return this.resources.fetch({ type: 'skus' }, `reserved_stocks/${_reservedStockId}/sku`, params, options);
    }
    async stock_reservations(reservedStockId, params, options) {
        const _reservedStockId = reservedStockId.id || reservedStockId;
        return this.resources.fetch({ type: 'stock_reservations' }, `reserved_stocks/${_reservedStockId}/stock_reservations`, params, options);
    }
    isReservedStock(resource) {
        return resource.type && (resource.type === ReservedStocks.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: ReservedStocks.TYPE } : { id: id.id, type: ReservedStocks.TYPE };
    }
    type() {
        return ReservedStocks.TYPE;
    }
}
ReservedStocks.TYPE = 'reserved_stocks';
export default ReservedStocks;
