// @commercelayer/sdk/lib/esm/resources/shipping_zones.js
import { ApiResource } from '../resource';
class ShippingZones extends ApiResource {
    async create(resource, params, options) {
        return this.resources.create({ ...resource, type: ShippingZones.TYPE }, params, options);
    }
    async update(resource, params, options) {
        return this.resources.update({ ...resource, type: ShippingZones.TYPE }, params, options);
    }
    async delete(id, options) {
        await this.resources.delete((typeof id === 'string') ? { id, type: ShippingZones.TYPE } : id, options);
    }
    async attachments(shippingZoneId, params, options) {
        const _shippingZoneId = shippingZoneId.id || shippingZoneId;
        return this.resources.fetch({ type: 'attachments' }, `shipping_zones/${_shippingZoneId}/attachments`, params, options);
    }
    async versions(shippingZoneId, params, options) {
        const _shippingZoneId = shippingZoneId.id || shippingZoneId;
        return this.resources.fetch({ type: 'versions' }, `shipping_zones/${_shippingZoneId}/versions`, params, options);
    }
    isShippingZone(resource) {
        return resource.type && (resource.type === ShippingZones.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: ShippingZones.TYPE } : { id: id.id, type: ShippingZones.TYPE };
    }
    type() {
        return ShippingZones.TYPE;
    }
}
ShippingZones.TYPE = 'shipping_zones';
export default ShippingZones;
