// @commercelayer/sdk/lib/esm/resources/tax_calculators.js
import { ApiResource } from '../resource';
class TaxCalculators extends ApiResource {
    async markets(taxCalculatorId, params, options) {
        const _taxCalculatorId = taxCalculatorId.id || taxCalculatorId;
        return this.resources.fetch({ type: 'markets' }, `tax_calculators/${_taxCalculatorId}/markets`, params, options);
    }
    async attachments(taxCalculatorId, params, options) {
        const _taxCalculatorId = taxCalculatorId.id || taxCalculatorId;
        return this.resources.fetch({ type: 'attachments' }, `tax_calculators/${_taxCalculatorId}/attachments`, params, options);
    }
    async versions(taxCalculatorId, params, options) {
        const _taxCalculatorId = taxCalculatorId.id || taxCalculatorId;
        return this.resources.fetch({ type: 'versions' }, `tax_calculators/${_taxCalculatorId}/versions`, params, options);
    }
    isTaxCalculator(resource) {
        return resource.type && (resource.type === TaxCalculators.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: TaxCalculators.TYPE } : { id: id.id, type: TaxCalculators.TYPE };
    }
    type() {
        return TaxCalculators.TYPE;
    }
}
TaxCalculators.TYPE = 'tax_calculators';
export default TaxCalculators;
