// @commercelayer/sdk/lib/esm/resources/tax_categories.js
import { ApiResource } from '../resource';
class TaxCategories extends ApiResource {
    async create(resource, params, options) {
        return this.resources.create({ ...resource, type: TaxCategories.TYPE }, params, options);
    }
    async update(resource, params, options) {
        return this.resources.update({ ...resource, type: TaxCategories.TYPE }, params, options);
    }
    async delete(id, options) {
        await this.resources.delete((typeof id === 'string') ? { id, type: TaxCategories.TYPE } : id, options);
    }
    async sku(taxCategoryId, params, options) {
        const _taxCategoryId = taxCategoryId.id || taxCategoryId;
        return this.resources.fetch({ type: 'skus' }, `tax_categories/${_taxCategoryId}/sku`, params, options);
    }
    async attachments(taxCategoryId, params, options) {
        const _taxCategoryId = taxCategoryId.id || taxCategoryId;
        return this.resources.fetch({ type: 'attachments' }, `tax_categories/${_taxCategoryId}/attachments`, params, options);
    }
    async versions(taxCategoryId, params, options) {
        const _taxCategoryId = taxCategoryId.id || taxCategoryId;
        return this.resources.fetch({ type: 'versions' }, `tax_categories/${_taxCategoryId}/versions`, params, options);
    }
    isTaxCategory(resource) {
        return resource.type && (resource.type === TaxCategories.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: TaxCategories.TYPE } : { id: id.id, type: TaxCategories.TYPE };
    }
    type() {
        return TaxCategories.TYPE;
    }
}
TaxCategories.TYPE = 'tax_categories';
export default TaxCategories;
