// @commercelayer/sdk/lib/cjs/resources/adjustments.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const resource_1 = require("../resource");
class Adjustments extends resource_1.ApiResource {
    async create(resource, params, options) {
        return this.resources.create({ ...resource, type: Adjustments.TYPE }, params, options);
    }
    async update(resource, params, options) {
        return this.resources.update({ ...resource, type: Adjustments.TYPE }, params, options);
    }
    async delete(id, options) {
        await this.resources.delete((typeof id === 'string') ? { id, type: Adjustments.TYPE } : id, options);
    }
    async versions(adjustmentId, params, options) {
        const _adjustmentId = adjustmentId.id || adjustmentId;
        return this.resources.fetch({ type: 'versions' }, `adjustments/${_adjustmentId}/versions`, params, options);
    }
    isAdjustment(resource) {
        return resource.type && (resource.type === Adjustments.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: Adjustments.TYPE } : { id: id.id, type: Adjustments.TYPE };
    }
    type() {
        return Adjustments.TYPE;
    }
}
Adjustments.TYPE = 'adjustments';
exports.default = Adjustments;
