// @commercelayer/sdk/lib/cjs/resources/carrier_accounts.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const resource_1 = require("../resource");
class CarrierAccounts extends resource_1.ApiResource {
    async market(carrierAccountId, params, options) {
        const _carrierAccountId = carrierAccountId.id || carrierAccountId;
        return this.resources.fetch({ type: 'markets' }, `carrier_accounts/${_carrierAccountId}/market`, params, options);
    }
    async attachments(carrierAccountId, params, options) {
        const _carrierAccountId = carrierAccountId.id || carrierAccountId;
        return this.resources.fetch({ type: 'attachments' }, `carrier_accounts/${_carrierAccountId}/attachments`, params, options);
    }
    async versions(carrierAccountId, params, options) {
        const _carrierAccountId = carrierAccountId.id || carrierAccountId;
        return this.resources.fetch({ type: 'versions' }, `carrier_accounts/${_carrierAccountId}/versions`, params, options);
    }
    isCarrierAccount(resource) {
        return resource.type && (resource.type === CarrierAccounts.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: CarrierAccounts.TYPE } : { id: id.id, type: CarrierAccounts.TYPE };
    }
    type() {
        return CarrierAccounts.TYPE;
    }
}
CarrierAccounts.TYPE = 'carrier_accounts';
exports.default = CarrierAccounts;
