// @commercelayer/sdk/lib/cjs/resources/checkout_com_gateways.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const resource_1 = require("../resource");
class CheckoutComGateways extends resource_1.ApiResource {
    async create(resource, params, options) {
        return this.resources.create({ ...resource, type: CheckoutComGateways.TYPE }, params, options);
    }
    async update(resource, params, options) {
        return this.resources.update({ ...resource, type: CheckoutComGateways.TYPE }, params, options);
    }
    async delete(id, options) {
        await this.resources.delete((typeof id === 'string') ? { id, type: CheckoutComGateways.TYPE } : id, options);
    }
    async payment_methods(checkoutComGatewayId, params, options) {
        const _checkoutComGatewayId = checkoutComGatewayId.id || checkoutComGatewayId;
        return this.resources.fetch({ type: 'payment_methods' }, `checkout_com_gateways/${_checkoutComGatewayId}/payment_methods`, params, options);
    }
    async versions(checkoutComGatewayId, params, options) {
        const _checkoutComGatewayId = checkoutComGatewayId.id || checkoutComGatewayId;
        return this.resources.fetch({ type: 'versions' }, `checkout_com_gateways/${_checkoutComGatewayId}/versions`, params, options);
    }
    async checkout_com_payments(checkoutComGatewayId, params, options) {
        const _checkoutComGatewayId = checkoutComGatewayId.id || checkoutComGatewayId;
        return this.resources.fetch({ type: 'checkout_com_payments' }, `checkout_com_gateways/${_checkoutComGatewayId}/checkout_com_payments`, params, options);
    }
    isCheckoutComGateway(resource) {
        return resource.type && (resource.type === CheckoutComGateways.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: CheckoutComGateways.TYPE } : { id: id.id, type: CheckoutComGateways.TYPE };
    }
    type() {
        return CheckoutComGateways.TYPE;
    }
}
CheckoutComGateways.TYPE = 'checkout_com_gateways';
exports.default = CheckoutComGateways;
