// @commercelayer/sdk/lib/cjs/resources/cleanups.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const resource_1 = require("../resource");
class Cleanups extends resource_1.ApiResource {
    async create(resource, params, options) {
        return this.resources.create({ ...resource, type: Cleanups.TYPE }, params, options);
    }
    async delete(id, options) {
        await this.resources.delete((typeof id === 'string') ? { id, type: Cleanups.TYPE } : id, options);
    }
    async events(cleanupId, params, options) {
        const _cleanupId = cleanupId.id || cleanupId;
        return this.resources.fetch({ type: 'events' }, `cleanups/${_cleanupId}/events`, params, options);
    }
    async versions(cleanupId, params, options) {
        const _cleanupId = cleanupId.id || cleanupId;
        return this.resources.fetch({ type: 'versions' }, `cleanups/${_cleanupId}/versions`, params, options);
    }
    isCleanup(resource) {
        return resource.type && (resource.type === Cleanups.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: Cleanups.TYPE } : { id: id.id, type: Cleanups.TYPE };
    }
    type() {
        return Cleanups.TYPE;
    }
}
Cleanups.TYPE = 'cleanups';
exports.default = Cleanups;
