// @commercelayer/sdk/lib/cjs/resources/exports.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const resource_1 = require("../resource");
class Exports extends resource_1.ApiResource {
    async create(resource, params, options) {
        return this.resources.create({ ...resource, type: Exports.TYPE }, params, options);
    }
    async delete(id, options) {
        await this.resources.delete((typeof id === 'string') ? { id, type: Exports.TYPE } : id, options);
    }
    async events(exportId, params, options) {
        const _exportId = exportId.id || exportId;
        return this.resources.fetch({ type: 'events' }, `exports/${_exportId}/events`, params, options);
    }
    isExport(resource) {
        return resource.type && (resource.type === Exports.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: Exports.TYPE } : { id: id.id, type: Exports.TYPE };
    }
    type() {
        return Exports.TYPE;
    }
}
Exports.TYPE = 'exports';
exports.default = Exports;
