// @commercelayer/sdk/lib/cjs/resources/imports.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const resource_1 = require("../resource");
class Imports extends resource_1.ApiResource {
    async create(resource, params, options) {
        return this.resources.create({ ...resource, type: Imports.TYPE }, params, options);
    }
    async delete(id, options) {
        await this.resources.delete((typeof id === 'string') ? { id, type: Imports.TYPE } : id, options);
    }
    async events(importId, params, options) {
        const _importId = importId.id || importId;
        return this.resources.fetch({ type: 'events' }, `imports/${_importId}/events`, params, options);
    }
    isImport(resource) {
        return resource.type && (resource.type === Imports.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: Imports.TYPE } : { id: id.id, type: Imports.TYPE };
    }
    type() {
        return Imports.TYPE;
    }
}
Imports.TYPE = 'imports';
exports.default = Imports;
