// @commercelayer/sdk/lib/cjs/resources/klarna_gateways.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const resource_1 = require("../resource");
class KlarnaGateways extends resource_1.ApiResource {
    async create(resource, params, options) {
        return this.resources.create({ ...resource, type: KlarnaGateways.TYPE }, params, options);
    }
    async update(resource, params, options) {
        return this.resources.update({ ...resource, type: KlarnaGateways.TYPE }, params, options);
    }
    async delete(id, options) {
        await this.resources.delete((typeof id === 'string') ? { id, type: KlarnaGateways.TYPE } : id, options);
    }
    async payment_methods(klarnaGatewayId, params, options) {
        const _klarnaGatewayId = klarnaGatewayId.id || klarnaGatewayId;
        return this.resources.fetch({ type: 'payment_methods' }, `klarna_gateways/${_klarnaGatewayId}/payment_methods`, params, options);
    }
    async versions(klarnaGatewayId, params, options) {
        const _klarnaGatewayId = klarnaGatewayId.id || klarnaGatewayId;
        return this.resources.fetch({ type: 'versions' }, `klarna_gateways/${_klarnaGatewayId}/versions`, params, options);
    }
    async klarna_payments(klarnaGatewayId, params, options) {
        const _klarnaGatewayId = klarnaGatewayId.id || klarnaGatewayId;
        return this.resources.fetch({ type: 'klarna_payments' }, `klarna_gateways/${_klarnaGatewayId}/klarna_payments`, params, options);
    }
    isKlarnaGateway(resource) {
        return resource.type && (resource.type === KlarnaGateways.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: KlarnaGateways.TYPE } : { id: id.id, type: KlarnaGateways.TYPE };
    }
    type() {
        return KlarnaGateways.TYPE;
    }
}
KlarnaGateways.TYPE = 'klarna_gateways';
exports.default = KlarnaGateways;
