// @commercelayer/sdk/lib/cjs/resources/order_amount_promotion_rules.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const resource_1 = require("../resource");
class OrderAmountPromotionRules extends resource_1.ApiResource {
    async create(resource, params, options) {
        return this.resources.create({ ...resource, type: OrderAmountPromotionRules.TYPE }, params, options);
    }
    async update(resource, params, options) {
        return this.resources.update({ ...resource, type: OrderAmountPromotionRules.TYPE }, params, options);
    }
    async delete(id, options) {
        await this.resources.delete((typeof id === 'string') ? { id, type: OrderAmountPromotionRules.TYPE } : id, options);
    }
    async versions(orderAmountPromotionRuleId, params, options) {
        const _orderAmountPromotionRuleId = orderAmountPromotionRuleId.id || orderAmountPromotionRuleId;
        return this.resources.fetch({ type: 'versions' }, `order_amount_promotion_rules/${_orderAmountPromotionRuleId}/versions`, params, options);
    }
    isOrderAmountPromotionRule(resource) {
        return resource.type && (resource.type === OrderAmountPromotionRules.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: OrderAmountPromotionRules.TYPE } : { id: id.id, type: OrderAmountPromotionRules.TYPE };
    }
    type() {
        return OrderAmountPromotionRules.TYPE;
    }
}
OrderAmountPromotionRules.TYPE = 'order_amount_promotion_rules';
exports.default = OrderAmountPromotionRules;
