// @commercelayer/sdk/lib/cjs/resources/paypal_gateways.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const resource_1 = require("../resource");
class PaypalGateways extends resource_1.ApiResource {
    async create(resource, params, options) {
        return this.resources.create({ ...resource, type: PaypalGateways.TYPE }, params, options);
    }
    async update(resource, params, options) {
        return this.resources.update({ ...resource, type: PaypalGateways.TYPE }, params, options);
    }
    async delete(id, options) {
        await this.resources.delete((typeof id === 'string') ? { id, type: PaypalGateways.TYPE } : id, options);
    }
    async payment_methods(paypalGatewayId, params, options) {
        const _paypalGatewayId = paypalGatewayId.id || paypalGatewayId;
        return this.resources.fetch({ type: 'payment_methods' }, `paypal_gateways/${_paypalGatewayId}/payment_methods`, params, options);
    }
    async versions(paypalGatewayId, params, options) {
        const _paypalGatewayId = paypalGatewayId.id || paypalGatewayId;
        return this.resources.fetch({ type: 'versions' }, `paypal_gateways/${_paypalGatewayId}/versions`, params, options);
    }
    async paypal_payments(paypalGatewayId, params, options) {
        const _paypalGatewayId = paypalGatewayId.id || paypalGatewayId;
        return this.resources.fetch({ type: 'paypal_payments' }, `paypal_gateways/${_paypalGatewayId}/paypal_payments`, params, options);
    }
    isPaypalGateway(resource) {
        return resource.type && (resource.type === PaypalGateways.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: PaypalGateways.TYPE } : { id: id.id, type: PaypalGateways.TYPE };
    }
    type() {
        return PaypalGateways.TYPE;
    }
}
PaypalGateways.TYPE = 'paypal_gateways';
exports.default = PaypalGateways;
