// @commercelayer/sdk/lib/cjs/resources/price_lists.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const resource_1 = require("../resource");
class PriceLists extends resource_1.ApiResource {
    async create(resource, params, options) {
        return this.resources.create({ ...resource, type: PriceLists.TYPE }, params, options);
    }
    async update(resource, params, options) {
        return this.resources.update({ ...resource, type: PriceLists.TYPE }, params, options);
    }
    async delete(id, options) {
        await this.resources.delete((typeof id === 'string') ? { id, type: PriceLists.TYPE } : id, options);
    }
    async prices(priceListId, params, options) {
        const _priceListId = priceListId.id || priceListId;
        return this.resources.fetch({ type: 'prices' }, `price_lists/${_priceListId}/prices`, params, options);
    }
    async attachments(priceListId, params, options) {
        const _priceListId = priceListId.id || priceListId;
        return this.resources.fetch({ type: 'attachments' }, `price_lists/${_priceListId}/attachments`, params, options);
    }
    async versions(priceListId, params, options) {
        const _priceListId = priceListId.id || priceListId;
        return this.resources.fetch({ type: 'versions' }, `price_lists/${_priceListId}/versions`, params, options);
    }
    isPriceList(resource) {
        return resource.type && (resource.type === PriceLists.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: PriceLists.TYPE } : { id: id.id, type: PriceLists.TYPE };
    }
    type() {
        return PriceLists.TYPE;
    }
}
PriceLists.TYPE = 'price_lists';
exports.default = PriceLists;
