// @commercelayer/sdk/lib/cjs/resources/price_frequency_tiers.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const resource_1 = require("../resource");
class PriceFrequencyTiers extends resource_1.ApiResource {
    async create(resource, params, options) {
        return this.resources.create({ ...resource, type: PriceFrequencyTiers.TYPE }, params, options);
    }
    async update(resource, params, options) {
        return this.resources.update({ ...resource, type: PriceFrequencyTiers.TYPE }, params, options);
    }
    async delete(id, options) {
        await this.resources.delete((typeof id === 'string') ? { id, type: PriceFrequencyTiers.TYPE } : id, options);
    }
    async price(priceFrequencyTierId, params, options) {
        const _priceFrequencyTierId = priceFrequencyTierId.id || priceFrequencyTierId;
        return this.resources.fetch({ type: 'prices' }, `price_frequency_tiers/${_priceFrequencyTierId}/price`, params, options);
    }
    async attachments(priceFrequencyTierId, params, options) {
        const _priceFrequencyTierId = priceFrequencyTierId.id || priceFrequencyTierId;
        return this.resources.fetch({ type: 'attachments' }, `price_frequency_tiers/${_priceFrequencyTierId}/attachments`, params, options);
    }
    async versions(priceFrequencyTierId, params, options) {
        const _priceFrequencyTierId = priceFrequencyTierId.id || priceFrequencyTierId;
        return this.resources.fetch({ type: 'versions' }, `price_frequency_tiers/${_priceFrequencyTierId}/versions`, params, options);
    }
    async events(priceFrequencyTierId, params, options) {
        const _priceFrequencyTierId = priceFrequencyTierId.id || priceFrequencyTierId;
        return this.resources.fetch({ type: 'events' }, `price_frequency_tiers/${_priceFrequencyTierId}/events`, params, options);
    }
    isPriceFrequencyTier(resource) {
        return resource.type && (resource.type === PriceFrequencyTiers.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: PriceFrequencyTiers.TYPE } : { id: id.id, type: PriceFrequencyTiers.TYPE };
    }
    type() {
        return PriceFrequencyTiers.TYPE;
    }
}
PriceFrequencyTiers.TYPE = 'price_frequency_tiers';
exports.default = PriceFrequencyTiers;
