// @commercelayer/sdk/lib/cjs/resources/price_tiers.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const resource_1 = require("../resource");
class PriceTiers extends resource_1.ApiResource {
    async price(priceTierId, params, options) {
        const _priceTierId = priceTierId.id || priceTierId;
        return this.resources.fetch({ type: 'prices' }, `price_tiers/${_priceTierId}/price`, params, options);
    }
    async attachments(priceTierId, params, options) {
        const _priceTierId = priceTierId.id || priceTierId;
        return this.resources.fetch({ type: 'attachments' }, `price_tiers/${_priceTierId}/attachments`, params, options);
    }
    async versions(priceTierId, params, options) {
        const _priceTierId = priceTierId.id || priceTierId;
        return this.resources.fetch({ type: 'versions' }, `price_tiers/${_priceTierId}/versions`, params, options);
    }
    isPriceTier(resource) {
        return resource.type && (resource.type === PriceTiers.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: PriceTiers.TYPE } : { id: id.id, type: PriceTiers.TYPE };
    }
    type() {
        return PriceTiers.TYPE;
    }
}
PriceTiers.TYPE = 'price_tiers';
exports.default = PriceTiers;
