// @commercelayer/sdk/lib/cjs/resources/shipping_method_tiers.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const resource_1 = require("../resource");
class ShippingMethodTiers extends resource_1.ApiResource {
    async shipping_method(shippingMethodTierId, params, options) {
        const _shippingMethodTierId = shippingMethodTierId.id || shippingMethodTierId;
        return this.resources.fetch({ type: 'shipping_methods' }, `shipping_method_tiers/${_shippingMethodTierId}/shipping_method`, params, options);
    }
    async attachments(shippingMethodTierId, params, options) {
        const _shippingMethodTierId = shippingMethodTierId.id || shippingMethodTierId;
        return this.resources.fetch({ type: 'attachments' }, `shipping_method_tiers/${_shippingMethodTierId}/attachments`, params, options);
    }
    async versions(shippingMethodTierId, params, options) {
        const _shippingMethodTierId = shippingMethodTierId.id || shippingMethodTierId;
        return this.resources.fetch({ type: 'versions' }, `shipping_method_tiers/${_shippingMethodTierId}/versions`, params, options);
    }
    isShippingMethodTier(resource) {
        return resource.type && (resource.type === ShippingMethodTiers.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: ShippingMethodTiers.TYPE } : { id: id.id, type: ShippingMethodTiers.TYPE };
    }
    type() {
        return ShippingMethodTiers.TYPE;
    }
}
ShippingMethodTiers.TYPE = 'shipping_method_tiers';
exports.default = ShippingMethodTiers;
