// @commercelayer/sdk/lib/cjs/resources/taxjar_accounts.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const resource_1 = require("../resource");
class TaxjarAccounts extends resource_1.ApiResource {
    async create(resource, params, options) {
        return this.resources.create({ ...resource, type: TaxjarAccounts.TYPE }, params, options);
    }
    async update(resource, params, options) {
        return this.resources.update({ ...resource, type: TaxjarAccounts.TYPE }, params, options);
    }
    async delete(id, options) {
        await this.resources.delete((typeof id === 'string') ? { id, type: TaxjarAccounts.TYPE } : id, options);
    }
    async markets(taxjarAccountId, params, options) {
        const _taxjarAccountId = taxjarAccountId.id || taxjarAccountId;
        return this.resources.fetch({ type: 'markets' }, `taxjar_accounts/${_taxjarAccountId}/markets`, params, options);
    }
    async attachments(taxjarAccountId, params, options) {
        const _taxjarAccountId = taxjarAccountId.id || taxjarAccountId;
        return this.resources.fetch({ type: 'attachments' }, `taxjar_accounts/${_taxjarAccountId}/attachments`, params, options);
    }
    async versions(taxjarAccountId, params, options) {
        const _taxjarAccountId = taxjarAccountId.id || taxjarAccountId;
        return this.resources.fetch({ type: 'versions' }, `taxjar_accounts/${_taxjarAccountId}/versions`, params, options);
    }
    async tax_categories(taxjarAccountId, params, options) {
        const _taxjarAccountId = taxjarAccountId.id || taxjarAccountId;
        return this.resources.fetch({ type: 'tax_categories' }, `taxjar_accounts/${_taxjarAccountId}/tax_categories`, params, options);
    }
    isTaxjarAccount(resource) {
        return resource.type && (resource.type === TaxjarAccounts.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: TaxjarAccounts.TYPE } : { id: id.id, type: TaxjarAccounts.TYPE };
    }
    type() {
        return TaxjarAccounts.TYPE;
    }
}
TaxjarAccounts.TYPE = 'taxjar_accounts';
exports.default = TaxjarAccounts;
