// @commercelayer/sdk/lib/cjs/resources/wire_transfers.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const resource_1 = require("../resource");
class WireTransfers extends resource_1.ApiResource {
    async create(resource, params, options) {
        return this.resources.create({ ...resource, type: WireTransfers.TYPE }, params, options);
    }
    async update(resource, params, options) {
        return this.resources.update({ ...resource, type: WireTransfers.TYPE }, params, options);
    }
    async delete(id, options) {
        await this.resources.delete((typeof id === 'string') ? { id, type: WireTransfers.TYPE } : id, options);
    }
    async order(wireTransferId, params, options) {
        const _wireTransferId = wireTransferId.id || wireTransferId;
        return this.resources.fetch({ type: 'orders' }, `wire_transfers/${_wireTransferId}/order`, params, options);
    }
    async versions(wireTransferId, params, options) {
        const _wireTransferId = wireTransferId.id || wireTransferId;
        return this.resources.fetch({ type: 'versions' }, `wire_transfers/${_wireTransferId}/versions`, params, options);
    }
    isWireTransfer(resource) {
        return resource.type && (resource.type === WireTransfers.TYPE);
    }
    relationship(id) {
        return ((id === null) || (typeof id === 'string')) ? { id, type: WireTransfers.TYPE } : { id: id.id, type: WireTransfers.TYPE };
    }
    type() {
        return WireTransfers.TYPE;
    }
}
WireTransfers.TYPE = 'wire_transfers';
exports.default = WireTransfers;
